---
title: "Annotation"
format: 
  html:
    link-external-newwindow: true
---

Next, we annotate the significant CpG sites (are they located in a promoter region or gene body?) and run a GO analysis to identify potentially important GO terms associated with the process. We repeat the same steps for all models run, where first I report the number of CpG sites across gene regions looking at all filtered CpG sites and then those that were significant in that specific model. Then, I annotated all the significant CpG sites for that model and using lifted over chicken genome, I report the genes that contain at least one significant CpG site. Then, I conduct a GO analysis by comparing the list of genes with a significant CpG site to the list of all genes with a filtered CpG site using GOrilla (which I only do for models where there are at least 50 significant CpG sites).

## Changing CpG sites

First, we look at the CpG sites that significantly change across the lekking season (@sec-changingcpg).

### Regions

```{r plot_setup, echo=F}
pacman::p_load(tidyverse, EnvStats, kableExtra, readxl)
sum_annotated <- read.csv(file="../results/tables/summary_regions_sig_cpgs_all.csv")
source("../scripts/plotting_theme.R")
sum_annotated$region <- factor(sum_annotated$region, levels = c("3' UTR", "5' UTR", "Downstream", "Upstream", "Gene body", "Exon", "Intron", "Promoter", "TSS"))

```

```{r plot_changing, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Time period")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of CpG sites that significantly change across the lekking period was `r sum_annotated$n_total[which(sum_annotated$model == "Time period")][1]`. It appears that changing CpG sites are present in functional regions more than expected.

### Genes

Which genes are these significant CpG sites located in?

```{r sig_genes_changing}
table_genes_changing <- read.csv("../results/tables/sig_gene_ids_time_period.csv")

table_genes_changing %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

```

### GO terms

Can we identify any significant GO terms associated with these genes using two unranked lists, one the target and one the background?

```{r sig_go_changing}
table_go_function_changing <- read_excel("../results/tables/go_time_period_function_2list.xlsx")

table_go_function_changing %>% select(-Genes) %>% filter(grepl("^GO", table_go_function_changing$`GO Term`)) %>% kbl() %>%  kable_classic_2()%>%  scroll_box(width = "99%", height = "200px")

table_go_process_changing <- read_excel("../results/tables/go_time_period_process_2list.xlsx")

table_go_process_changing %>% select(-Genes) %>% filter(grepl("^GO", table_go_process_changing$`GO Term`)) %>% kbl() %>%  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

```
## CpG sites associated with reproductive effort

### Regions

We can do the same with the CpG sites significantly associated with reproductive effort (lek attendance, fighting rate and lek centrality).

```{r plot_attend, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Attendance")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with lek attendance were `r sum_annotated$n_total[which(sum_annotated$model == "Attendance")][1]`.

```{r plot_fight, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Fighting rate")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with fighting rate were `r sum_annotated$n_total[which(sum_annotated$model == "Fighting rate")][1]`.

```{r plot_dist, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Centrality")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with lek centrality were `r sum_annotated$n_total[which(sum_annotated$model == "Centrality")][1]`.

### Genes

Which genes are these significant CpG sites located in?

```{r sig_genes_effort}
## attend
table_genes_attend <- read.csv("../results/tables/sig_gene_ids_attend.csv")

table_genes_attend %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

## fight
table_genes_fight <- read.csv("../results/tables/sig_gene_ids_fight.csv")

table_genes_fight %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

## dist
table_genes_dist <- read.csv("../results/tables/sig_gene_ids_dist.csv")

table_genes_dist %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

```

## CpG sites associated with changing physiology

### Regions 

Below you can find similar plots for CpG sites significantly associated with changes in physiology (mass, HCT, IgG, Microfilaria, Trypanosoma)

```{r plot_mass, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Delta body mass")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with changing body mass were `r sum_annotated$n_total[which(sum_annotated$model == "Delta body mass")][1]`.

```{r plot_hct, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Delta HCT")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with delta HCT were `r sum_annotated$n_total[which(sum_annotated$model == "Delta HCT")][1]`.

```{r plot_igg, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Delta IgG")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with delta IgG were `r sum_annotated$n_total[which(sum_annotated$model == "Delta IgG")][1]`.

```{r plot_microf, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Delta Microfilaria spp.")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with delta Microfilaria were `r sum_annotated$n_total[which(sum_annotated$model == "Delta Microfilaria spp.")][1]`.

```{r plot_trypa, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Delta Trypanosoma spp.")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with delta Trypanosoma were `r sum_annotated$n_total[which(sum_annotated$model == "Delta Trypanosoma spp.")][1]`.

### Genes


Which genes are these significant CpG sites located in?

```{r sig_genes_physio}
## delta mass
table_genes_mass <- read.csv("../results/tables/sig_gene_ids_mass.csv")

table_genes_mass %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

## delta hct
table_genes_hct <- read.csv("../results/tables/sig_gene_ids_htc.csv")

table_genes_hct %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

## igg
table_genes_igg <- read.csv("../results/tables/sig_gene_ids_igg.csv")

table_genes_igg %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

## microfilaria
table_genes_microf <- read.csv("../results/tables/sig_gene_ids_microf.csv")

table_genes_microf %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

## trypa
table_genes_trypa <- read.csv("../results/tables/sig_gene_ids_trypa.csv")

table_genes_trypa %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

```

## CpG sites associated with AMS

### Regions

Lastly, here you can find the plots for CpG sites significantly associated with annual mating success (AMS)

```{r plot_ams, echo=T, warning=F}
sub <- subset(sum_annotated, model == "All" | model == "Annual mating success")
ggplot(sub, aes(x = region, y = perc)) + 
    geom_bar(stat="identity", fill = clrs[3]) + 
    facet_wrap(~model, ncol=2) + ylim(0,100) + geom_text(label=paste0("N = ", sub$n), hjust=-0.2, family = "Arial", size = 4) +
    labs(x = "Region", y = "Percentage CpG sites") + coord_flip()

```

The total number of changing CpG sites that were significantly associated with AMS were `r sum_annotated$n_total[which(sum_annotated$model == "Annual mating success")][1]`.

### Genes

Which genes are these significant CpG sites located in?

```{r sig_genes_ams}
## AMS
table_genes_ams <- read.csv("../results/tables/sig_gene_ids_AMS.csv")

table_genes_ams %>% kbl() %>%
  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

```

### GO terms

```{r sig_go_ams}
table_go_function_ams <- read_excel("../results/tables/go_ams_function_2list.xlsx")

table_go_function_ams %>% select(-Genes) %>% filter(grepl("^GO", table_go_function_ams$`GO Term`)) %>% kbl() %>%  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

table_go_process_ams <- read_excel("../results/tables/go_ams_process_2list.xlsx")

table_go_process_ams %>% select(-Genes) %>% filter(grepl("^GO", table_go_process_ams$`GO Term`)) %>% kbl() %>%  kable_classic_2() %>%  scroll_box(width = "99%", height = "200px")

```
