---
title: "Physio-associated CpG sites"
format: 
  html:
    link-external-newwindow: true
    fig-cap-location: top
---

**Which changes in CpG site methylation are associated with changes in physiology?**

Similar to the previous section, we next test which CpG site changes are significantly associated with changes in physiology. Physiological measures that we focus on include parasites (*Microfilaria spp.* and *Trypanosoma spp.*), body mass, IgG and HCT.

Again, we build a GLMM per CpG site to identify CpG sites whose DNA methylation change is associated with changes in physiology. For each CpG site for each trait, we build a model that predicts the changes DNA methylation, while fitting the change in physiology as a fixed effect. Here too, we run two models in parallel: one that does and one does that not include the fixed effect of pre-lekking methylation %.

## Prepare the data {#sec-physio}

```{r delta_meth, echo=T, eval=F}
#| code-fold: true
### load packages
pacman::p_load(tidyverse, data.table, tibble, 
               parallel, performance, lmerTest, tidystats)

### load data

load(file="results/modeloutput/changing/changing_sites_glmer.RData")

### load phenotypic data

load("data/phenotypes/fulldata_complete_epi_withdates.RData")

load("data/phenotypes/pheno_dif_prepost.RData")  ## differences in physiology

### combine delta methylation data with site and behaviour info

delta_meth <- left_join(delta_meth, unique(all_pheno_epi[,c("id", "year", "site", "Core")], by = c("id", "year")))

### z-transform the traits before the model
prepost_dif$mass_dif_scl <- scale(prepost_dif$mass_dif)
prepost_dif$microf_dif_scl <- scale(prepost_dif$microf_dif)
prepost_dif$trypa_dif_scl <- scale(prepost_dif$trypa_dif)
prepost_dif$ig_dif_scl <- scale(prepost_dif$ig_dif)
prepost_dif$hct_dif_scl <- scale(prepost_dif$hct_dif)

### combine data with pre-post delta physio numbers
delta_meth <- left_join(delta_meth, unique(prepost_dif[,c("id", "year", "mass_dif", "microf_dif", "trypa_dif", "ig_dif", "hct_dif",
                        "mass_dif_scl", "microf_dif_scl", "trypa_dif_scl", "ig_dif_scl", "hct_dif_scl")], by = c("id", "year")))



```

## Function to run the model

Below you'll find the function ran per trait per CpG site, where the LMM ran follows this structure:

`lmerTest::lmer(delta_meth ~ scale(delta_trait) + methperc_pre + (1|site/id))`

This model is identical to the one presented in @sec-model-trait, except that the trait in question is also a calculation of the difference in the trait (delta; e.g. delta mass) rather than just one measure (e.g. attendance).

## Run the model per trait

Here too, we only include CpG sites if they have at least 20 data points, which is therefore done for physiological trait separately. I'll give the code for one trait only (delta mass) but the exact same is done for the rest.

```{r run_model_example, echo=T, eval=F}
#| code-fold: true
### run the model per trait
source("scripts/function_models.R") 

### only select cpg sites with enough data
delta_meth_n_mass <- delta_meth %>% group_by(chr_pos) %>% filter(!is.na(delta_meth)& !is.na(mass_dif)) %>% tally()
delta_meth_n_mass <- subset(delta_meth_n_mass, n > 20)
 
delta_meth_sub_mass <- subset(delta_meth, chr_pos %in% delta_meth_n_mass$chr_pos)
length(unique(delta_meth_sub_mass$chr_pos)) #808 sites

delta_meth_sub_mass_ls <- delta_meth_sub_mass %>% group_split(chr_pos)

### mass

### no pre ####
m_mass_no_pre <- parallel::mclapply(delta_meth_sub_mass_ls, function_model_delta_pheno, parameter="mass_dif", pre="no_control", mc.cores=4)
m_mass_no_pre_out <- function_process_model(m_mass_no_pre, dir_plots = "plots/model_out/physio/mass", dir_data = "results/modeloutput/physio",name_file = "mass_dif_no_pre", pretty_name = "Delta mass", filter_disp=FALSE) 

### with pre ####
m_mass_with_pre <- parallel::mclapply(delta_meth_sub_mass_ls, function_model_delta_pheno, parameter="mass_dif", pre="control", mc.cores=4)
m_mass_with_pre_out <- function_process_model(m_mass_with_pre, dir_plots = "plots/model_out/physio/mass", dir_data = "results/modeloutput/physio",name_file = "mass_dif_with_pre", pretty_name = "Delta mass", filter_disp=FALSE) 
```

### Delta mass

808 CpG sites had enough phenotypic and epigenetic data to run the body mass model (after filtering for n \> 20).

#### With pre-lekking

![Histogram dispersion and p-value: delta mass with pre-methylation](../plots/model_out/physio/mass/hist_raw_disp_pval_mass_dif_with_pre.png)

![Volcano plot: delta mass with pre-methylation](../plots/model_out/physio/mass/volcano_mass_dif_with_pre.png)

![Raw data: delta mass with pre-methylation CpG 1](../plots/model_out/physio/mass/mass_with_pre_1.png)

![Raw data: delta mass with pre-methylation CpG 2](../plots/model_out/physio/mass/mass_with_pre_2.png)

CpG site "ScEsiA3_17616\_\_HRSCAF_20466_3956821" is located downstream from gene IDs ANN28634 (similar to BPI) and ANN28635 (similar to ID1, closer).

::: {.callout-note collapse="true"}
BPI: BPI (Bactericidal Permeability Increasing Protein) is a Protein Coding gene. Diseases associated with BPI include Meningococcemia and Reactive Arthritis. Among its related pathways are Innate Immune System and Toll Like Receptor 3 (TLR3) Cascade. Gene Ontology (GO) annotations related to this gene include lipid binding and lipopolysaccharide binding. An important paralog of this gene is LBP.

ID1: The protein encoded by this gene is a helix-loop-helix (HLH) protein that can form heterodimers with members of the basic HLH family of transcription factors. The encoded protein has no DNA binding activity and therefore can inhibit the DNA binding and transcriptional activation ability of basic HLH proteins with which it interacts. This protein may play a role in cell growth, senescence, and differentiation. Two transcript variants encoding different isoforms have been found for this gene. \[provided by RefSeq, Jul 2008\]
:::

CpG site "ScEsiA3_18278\_\_HRSCAF_21663_133779007" is located in the exon of ANN06029 (similar to Adck2) and upstream from ANN06028 and ANN06030 (both unknown functions). This CpG site is the same that is significantly associated with attendance rate where we include pre-lekking methylation % in the model (!).

#### Without pre-lekking

![Histogram dispersion and p-value: delta mass without pre-methylation](../plots/model_out/physio/mass/hist_raw_disp_pval_mass_dif_no_pre.png)

![Volcano plot: delta mass without pre-methylation](../plots/model_out/physio/mass/volcano_mass_dif_no_pre.png)

![Raw data: delta mass without pre-methylation CpG 1](../plots/model_out/physio/mass/mass_no_pre_1.png)

![Raw data: delta mass without pre-methylation CpG 2](../plots/model_out/physio/mass/mass_no_pre_2.png)

![Raw data: delta mass without pre-methylation CpG 3](../plots/model_out/physio/mass/mass_no_pre_3.png)

![Raw data: delta mass without pre-methylation CpG 4](../plots/model_out/physio/mass/mass_no_pre_4.png)

![Raw data: delta mass without pre-methylation CpG 5](../plots/model_out/physio/mass/mass_no_pre_5.png)

One of the significant 5 CpG sites overlaps with the model that does include pre-methylation %.

CpG site "ScEsiA3_15486\_\_HRSCAF_17393_53786697" is located in an intron (gene ID ANN20472 similar to Six4).

::: {.callout-note collapse="true"}
Six4: This gene encodes a member of the homeobox family, subfamily SIX. The drosophila homolog is a nuclear homeoprotein required for eye development. Studies in mouse show that this gene product functions as a transcription factor, and may have a role in the differentiation or maturation of neuronal cells. \[provided by RefSeq, May 2010\]
:::

CpG site "ScEsiA3_17616\_\_HRSCAF_20466_3956821" is located in a downstream region (gene ID ANN28634, similar to BPI, and ANN28635 similar to ID1). (see above)

CpG site "ScEsiA3_18752\_\_HRSCAF_22883_4031831" is located in an intron (gene ID ANN31264 similar to TET3).

::: {.callout-note collapse="true"}
TET3: Enables methyl-CpG binding activity and zinc ion binding activity. Involved in histone H3-K4 trimethylation; positive regulation of transcription by RNA polymerase II; and protein O-linked glycosylation. Predicted to be located in cytoplasm and male pronucleus. Predicted to be active in nucleus. Biomarker of esophagus squamous cell carcinoma. \[provided by Alliance of Genome Resources, Apr 2022\]
:::

CpG site "ScEsiA3_21976\_\_HRSCAF_26926_4403331" is located upstream from ANN30833 (similar to IFI27) and downstream from gene ID ANN30832 (similar to Kiaa0319l, closer).

::: {.callout-note collapse="true"}
IFI27: Enables RNA polymerase II-specific DNA-binding transcription factor binding activity; identical protein binding activity; and lamin binding activity. Involved in several processes, including cellular protein metabolic process; defense response to other organism; and extrinsic apoptotic signaling pathway. Acts upstream of or within negative regulation of transcription by RNA polymerase II and regulation of protein export from nucleus. Located in mitochondrial membrane and nuclear inner membrane. [provided by Alliance of Genome Resources, Apr 2022]

Kiaa0319l: Predicted to act upstream of or within several processes, including flagellated sperm motility; proacrosomal vesicle fusion; and receptor-mediated endocytosis of virus by host cell. Located in Golgi apparatus; cytoplasmic vesicle; and nucleolus. \[provided by Alliance of Genome Resources, Apr 2022\]
:::

### Microfilaria spp.

149 CpG sites had enough phenotypic and epigenetic data to run the Microfilaria model (after filtering for n \> 20).

#### With pre-lekking

![Histogram dispersion and p-value: delta Microfilaria spp. with pre-methylation](../plots/model_out/physio/microf/hist_raw_disp_pval_microf_dif_with_pre.png)

![Volcano plot: delta Microfilaria spp. with pre-methylation](../plots/model_out/physio/microf/volcano_microf_dif_with_pre.png)

![Raw data: delta Microfilaria spp. with pre-methylation CpG 1](../plots/model_out/physio/microf/microf_with_pre_1.png)

![Raw data: delta Microfilaria spp. with pre-methylation CpG 2](../plots/model_out/physio/microf/microf_with_pre_2.png)

![Raw data: delta Microfilaria spp. with pre-methylation CpG 3](../plots/model_out/physio/microf/microf_with_pre_3.png)

CpG site "ScEsiA3_16766\_\_HRSCAF_19082_28300178" was unannotated and CpG site "ScEsiA3_16771\_\_HRSCAF_19097_2512626" is located in an exon of gene ID ANN30965 (similar to JAM3).

::: {.callout-note collapse="true"}
JAM3: Tight junctions represent one mode of cell-to-cell adhesion in epithelial or endothelial cell sheets, forming continuous seals around cells and serving as a physical barrier to prevent solutes and water from passing freely through the paracellular space. The protein encoded by this immunoglobulin superfamily gene member is localized in the tight junctions between high endothelial cells. Unlike other proteins in this family, the this protein is unable to adhere to leukocyte cell lines and only forms weak homotypic interactions. The encoded protein is a member of the junctional adhesion molecule protein family and acts as a receptor for another member of this family. A mutation in an intron of this gene is associated with hemorrhagic destruction of the brain, subependymal calcification, and congenital cataracts. Alternative splicing results in multiple transcript variants.\[provided by RefSeq, Apr 2011\]
:::

CpG site "ScEsiA3_21978\_\_HRSCAF_26928_6268951" is located in a TSS (gene id ANN24632 similar to PPP1R1B) and upstream region (gene ID ANN24631 similar to STARD3).

::: {.callout-note collapse="true"}
PPP1R1B: This gene encodes a bifunctional signal transduction molecule. Dopaminergic and glutamatergic receptor stimulation regulates its phosphorylation and function as a kinase or phosphatase inhibitor. As a target for dopamine, this gene may serve as a therapeutic target for neurologic and psychiatric disorders. Multiple transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Oct 2011]

STARD3: This gene encodes a member of a subfamily of lipid trafficking proteins that are characterized by a C-terminal steroidogenic acute regulatory domain and an N-terminal metastatic lymph node 64 domain. The encoded protein localizes to the membranes of late endosomes and may be involved in exporting cholesterol. Alternative splicing results in multiple transcript variants.\[provided by RefSeq, Oct 2009\]
:::

#### Without pre-lekking

![Histogram dispersion and p-value: delta Microfilaria spp. without pre-methylation](../plots/model_out/physio/microf/hist_raw_disp_pval_microf_dif_no_pre.png)

![Volcano plot: delta Microfilaria spp. without pre-methylation](../plots/model_out/physio/microf/volcano_microf_dif_no_pre.png)

![Raw data: delta Microfilaria spp. without pre-methylation CpG 1](../plots/model_out/physio/microf/microf_no_pre_1.png)

This is not the same site that was significant in the model with pre-lekking methylation.

This CpG site was unannotated.

### Trypanosoma spp.

164 CpG sites had enough phenotypic and epigenetic data to run the Trypanosoma model (after filtering for n \> 20).

#### With pre-lekking

![Histogram dispersion and p-value: delta Trypanosoma spp. with pre-methylation](../plots/model_out/physio/trypa/hist_raw_disp_pval_trypa_dif_with_pre.png)

![Volcano plot: delta trypailaria spp. with pre-methylation](../plots/model_out/physio/trypa/volcano_trypa_dif_with_pre.png)

#### Without pre-lekking

![Histogram dispersion and p-value: delta trypailaria spp. without pre-methylation](../plots/model_out/physio/trypa/hist_raw_disp_pval_trypa_dif_no_pre.png)

![Volcano plot: delta trypailaria spp. without pre-methylation](../plots/model_out/physio/trypa/volcano_trypa_dif_no_pre.png)

### IgG

775 CpG sites had enough phenotypic and epigenetic data to run the IgG model (after filtering for n \> 20).

#### With pre-lekking

![Histogram dispersion and p-value: delta IgG with pre-methylation](../plots/model_out/physio/ig/hist_raw_disp_pval_ig_dif_with_pre.png)

![Volcano plot: delta IgG with pre-methylation](../plots/model_out/physio/ig/volcano_ig_dif_with_pre.png)

![Raw data: delta IgG with pre-methylation CpG 1](../plots/model_out/physio/ig/ig_with_pre_1.png)

![Raw data: delta IgG with pre-methylation CpG 2](../plots/model_out/physio/ig/ig_with_pre_2.png)

Both CpG site "ScEsiA3_21979\_\_HRSCAF_26929_1282292" (gene ID ANN27654 similar to NFIC) and "ScEsiA3_21979\_\_HRSCAF_26929_1282301" (gene ID ANN27654 similar to NFIC) are located in upstream regions. The first one is the same site that was associated with centrality!

#### Without pre-lekking

![Histogram dispersion and p-value: delta IgG without pre-methylation](../plots/model_out/physio/ig/hist_raw_disp_pval_ig_dif_no_pre.png)

![Volcano plot: delta IgG without pre-methylation](../plots/model_out/physio/ig/volcano_ig_dif_no_pre.png)

![Raw data: delta IgG without pre-methylation CpG 1](../plots/model_out/physio/ig/ig_no_pre_1.png)

![Raw data: delta IgG without pre-methylation CpG 2](../plots/model_out/physio/ig/ig_no_pre_2.png)

These were the exact same CpG sites that are significant compared to the model that did include pre-methylation levels.

### HCT

146 CpG sites had enough phenotypic and epigenetic data to run the HCT model (after filtering for n \> 20).

#### With pre-lekking

![Histogram dispersion and p-value: delta HCT with pre-methylation](../plots/model_out/physio/hct/hist_raw_disp_pval_hct_dif_with_pre.png)

![Volcano plot: delta HCT with pre-methylation](../plots/model_out/physio/hct/volcano_hct_dif_with_pre.png)

![Raw data: delta HCT with pre-methylation CpG 1](../plots/model_out/physio/hct/hct_with_pre_1.png)

CpG site "ScEsiA3_15486\_\_HRSCAF_17393_15878140" is located in a promoter region (gene ID ANN19643 similar to GAL: Galanin peptides (C. japonica)).

::: {.callout-note collapse="true"}
GAL: This gene encodes a neuroendocrine peptide that is widely expressed in the central and peripheral nervous systems and also the gastrointestinal tract, pancreas, adrenal gland and urogenital tract. The encoded protein is a precursor that is proteolytically processed to generate two mature peptides: galanin and galanin message-associated peptide (GMAP). Galanin has diverse physiological functions including nociception, feeding and energy homeostasis, osmotic regulation and water balance. GMAP has been demonstrated to possess antifungal activity and hypothesized to be part of the innate immune system. \[provided by RefSeq, Jul 2015\]
:::

#### Without pre-lekking

![Histogram dispersion and p-value: delta HCT without pre-methylation](../plots/model_out/physio/hct/hist_raw_disp_pval_hct_dif_no_pre.png)

![Volcano plot: delta HCT without pre-methylation](../plots/model_out/physio/hct/volcano_hct_dif_no_pre.png)
