<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>The role of epigenetics in regulating life history trade-offs - 3&nbsp; Whole genome bisulphite sequencing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../qmd/3_explore.html" rel="next">
<link href="../qmd/1_intro_hypothesis.html" rel="prev">
<link href="../img/grouse_silhouette.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The role of epigenetics in regulating life history trade-offs</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../qmd/2_wgbs.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole genome bisulphite sequencing</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Overview</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/1_intro_hypothesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction and hypotheses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/2_wgbs.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole genome bisulphite sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/3_explore.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Data exploration</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/4_filter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/5_changing_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/6_effort_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/7_physio_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/8_fitness_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Fitness effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/9_mutual_sites.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Mutual CpG sites</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#trim-align-call-methylation" id="toc-trim-align-call-methylation" class="nav-link active" data-scroll-target="#trim-align-call-methylation"><span class="header-section-number">4</span> Trim, align, call methylation</a>
  <ul class="collapse">
  <li><a href="#plot-the-result" id="toc-plot-the-result" class="nav-link" data-scroll-target="#plot-the-result"><span class="header-section-number">7.2</span> Plot the result</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole genome bisulphite sequencing</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Prior to analysing our epiGBS data, which uses a reduced representation approach, we characterised CpG methylation across the genome. More specifically, because epiGBS targets CpG islands which tend to cluster around promoter regions, we wanted to characterise CpG methylation levels around TSS/promoter regions.</p>
<section id="trim-align-call-methylation" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Trim, align, call methylation</h1>
<p>We used WGBS for one sample (E197) and trimmed the reads.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">trim_galore</span> <span class="at">--paired</span> data/E197_R1.fq.gz data/E197_R2.fq.gz <span class="at">--quality</span> 20 <span class="at">--fastqc</span> <span class="at">--illumina</span> <span class="at">--stringency</span> 3 <span class="at">--gzip</span> <span class="at">--length</span> 20 <span class="at">--trim-n</span> <span class="at">--cores</span> 8 <span class="at">-o</span> ./data/processed/trim/ <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span> <span class="fu">tee</span> log/trim.log</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next, we prepare the black grouse reference genome for aligning bisulphite-treated reads (using a container).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /vol/cluster-data/rchen/wgbs</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="va">WGBS_SCRATCH</span><span class="op">=</span><span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>/scratch/</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/vol/apptainer/bin/apptainer</span> run <span class="at">--cleanenv</span> <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fakeroot</span> <span class="at">--overlay</span> <span class="va">${WGBS_SCRATCH}</span> <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bind</span> <span class="va">${WGBS_SCRATCH}</span>/tmp,<span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>,/vol/cluster-data/rchen/wgbs/data/genome <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--env</span> PYTHONNOUSERSITE=1 <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  /vol/cluster-data/rchen/epi/epigbs2/singularity/bismark_0_24.sif <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  bismark_genome_preparation data/genome/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now that the genome is ready, we align the reads to the prepared reference genome.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/vol/apptainer/bin/apptainer</span> run <span class="at">--cleanenv</span> <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fakeroot</span> <span class="at">--overlay</span> <span class="va">${WGBS_SCRATCH}</span> <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bind</span> <span class="va">${WGBS_SCRATCH}</span>/tmp,<span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>,/vol/cluster-data/rchen/wgbs/data/genome <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--env</span> PYTHONNOUSERSITE=1 <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  /vol/cluster-data/rchen/epi/epigbs2/singularity/bismark_0_24.sif <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  bismark <span class="at">-p</span> 8 <span class="at">--parallel</span> 4 <span class="at">--un</span> <span class="at">--ambiguous</span> <span class="at">--genome</span> data/genome/ <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-1</span> data/processed/trim/E197_R1_val_1.fq.gz <span class="at">-2</span> data/processed/trim/E197_R2_val_2.fq.gz <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> data/processed/alignment <span class="at">--rg_tag</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We de-duplicate the aligned sequences with bismark</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/vol/apptainer/bin/apptainer</span> run <span class="at">--cleanenv</span> <span class="dt">\</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fakeroot</span> <span class="at">--overlay</span> <span class="va">${WGBS_SCRATCH}</span> <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bind</span> <span class="va">${WGBS_SCRATCH}</span>/tmp,<span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span> <span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--env</span> PYTHONNOUSERSITE=1 <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  /vol/cluster-data/rchen/epi/epigbs2/singularity/bismark_0_24.sif <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  deduplicate_bismark E197_R1_val_1_bismark_bt2_pe.bam </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Lastly, we call methylation using bismark.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/vol/apptainer/bin/apptainer</span> run <span class="at">--cleanenv</span> <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fakeroot</span> <span class="at">--overlay</span> <span class="va">${WGBS_SCRATCH}</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bind</span> <span class="va">${WGBS_SCRATCH}</span>/tmp,<span class="va">$(</span><span class="bu">pwd</span><span class="va">)</span>,/vol/cluster-data/rchen/wgbs/data/genome/ <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--env</span> PYTHONNOUSERSITE=1 <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  /vol/cluster-data/rchen/epi/epigbs2/singularity/bismark_0_24.sif <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  bismark_methylation_extractor <span class="at">--multicore</span> 4 <span class="at">--ignore</span> 4 <span class="at">--ignore_r2</span> 4 <span class="at">-p</span> <span class="at">--CX</span> <span class="at">--no_overlap</span> <span class="at">--scaffolds</span> <span class="at">--report</span> <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">--cutoff</span> 10 <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--bedGraph</span> <span class="at">--cytosine_report</span> <span class="at">--comprehensive</span> <span class="at">--gzip</span> <span class="at">--merge_non_CpG</span> <span class="at">--genome_folder</span> data/ref/ data/processed/deduplicated/E197_R1_val_1_bismark_bt2_pe.deduplicated.bam <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">-o</span> data/processed/methcall</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p># Downstream analysis</p>
<p>The last command generateds the file <code>methcall_report.CX_report.txt.gz</code> containing methylation calls. We further process this file in R to 1) annotate the CpG sites, 2) calculate average CpG methylation within gene regions across genes, using a sliding window approach. Next, we plot the result.</p>
<p>## Annotate CpG sites</p>
<p>To characterise methylation levels across different gene regions, we calculated the mean methylation level across genes for four regions: the transcription start site (TSS; 300 bp upstream to 50 bp downstream of the annotated starting position of each gene (Laine et al., 2016)), the gene body, upstream regions (10 kb upstream from the gene body) and downstream regions (10 kb downstream from the gene body).</p>
<p>```{r meth, echo=T, eval=F} #| code-fold: true</p>
<section id="packages" class="level4" data-number="4.0.0.1">
<h4 data-number="4.0.0.1" class="anchored" data-anchor-id="packages"><span class="header-section-number">4.0.0.1</span> packages</h4>
<p>pacman::p_load(data.table, genomation, GenomicFeatures, rtracklayer, GenomicRanges, windowscanr, dplyr, ggplot2, cowplot, assertthat)</p>
</section>
<section id="load-data" class="level4" data-number="4.0.0.2">
<h4 data-number="4.0.0.2" class="anchored" data-anchor-id="load-data"><span class="header-section-number">4.0.0.2</span> load data</h4>
<p>meth &lt;- fread(“data/processed/methcall/methcall_report.CX_report.txt.gz”)</p>
</section>
<section id="change-symbols-in-scaffold-names" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> change symbols in scaffold names</h1>
<p>meth<span class="math inline">\(V1 &lt;- gsub(";", "__", meth\)</span>V1) meth<span class="math inline">\(V1 &lt;- gsub("=", "_", meth\)</span>V1)</p>
</section>
<section id="name-columns" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> name columns</h1>
<p>names(meth) &lt;- c(“chr”, “start”, “strand”, “nC”, “nT”, “dicontext”, “tricontext”)</p>
</section>
<section id="add-end-column-for-ranges" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> add ‘end’ column for ranges</h1>
<p>meth<span class="math inline">\(end &lt;- meth\)</span>start</p>
<p>cpg &lt;- subset(meth, dicontext == “CG”)</p>
<section id="annotation" class="level5" data-number="7.0.0.0.1">
<h5 data-number="7.0.0.0.1" class="anchored" data-anchor-id="annotation"><span class="header-section-number">7.0.0.0.1</span> Annotation</h5>
</section>
<section id="make-a-function-to-annotate-dataframe" class="level3" data-number="7.0.1">
<h3 data-number="7.0.1" class="anchored" data-anchor-id="make-a-function-to-annotate-dataframe"><span class="header-section-number">7.0.1</span> make a function to annotate dataframe</h3>
<p>annotate_methfile &lt;- function(long_df, annotation_dir, regions){ pacman::p_load(genomation, GenomicFeatures, rtracklayer, GenomicRanges, tidyverse)</p>
<p>### annotate df accordingly and add region name long_df<span class="math inline">\(end &lt;- long_df\)</span>start df_gr &lt;- as(long_df, “GRanges”) # convert to GRanges object</p>
<p>#more complicated than just subsetting, want to include gene information too so need the overlap annotated &lt;- data.frame()</p>
<p>for (i in regions){ # loop over specified regions #load in gff file region &lt;- unique(gffToGRanges(paste0(annotation_dir, i, “.gff3”))) region &lt;- as(region, “GRanges”) #find overlaps in df and gene region, because can have multiple hits include all hits &lt;- findOverlaps(df_gr, region) idx_c &lt;- queryHits(hits) idx_region &lt;- subjectHits(hits)</p>
<pre><code>#get values from overlaps for cpg and region separately
values_region &lt;- data.frame(chr=region@seqnames[idx_region],
                            start = region@ranges@start[idx_region], 
                            end= region@ranges@start[idx_region] + region@ranges@width[idx_region], 
                            gene_id = mcols(region)$gene_id[idx_region],
                            region = i)

values_cytosines &lt;- data.frame(pos = df_gr@ranges@start[idx_c], 
                               nC = mcols(df_gr)$nC[idx_c], 
                               nT = mcols(df_gr)$nT[idx_c])

annotated_region &lt;- cbind(values_cytosines, values_region) #combine
annotated &lt;- rbind(annotated, annotated_region) #add to full df</code></pre>
<p>} annotated<span class="math inline">\(region &lt;- as.factor(annotated\)</span>region)</p>
<p>return(annotated) }</p>
</section>
<section id="annotate-cpg-and-ch-file" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="annotate-cpg-and-ch-file"><span class="header-section-number">7.1</span> Annotate CpG and CH file</h2>
<p>cpg_annotate &lt;- annotate_methfile(long_df = cpg, annotation_dir = “/vol/cluster-data/rchen/git/grouse-annotation/output/”, regions = c(“TSS”, “genes”, “upstream”, “downstream”))</p>
<p>cpg_annotate<span class="math inline">\(methperc &lt;- cpg_annotate\)</span>nC / (cpg_annotate<span class="math inline">\(nC + cpg_annotate\)</span>nT) * 100</p>
<p>save(cpg_annotate, file = “data/processed/annotated_cpg.RData”)</p>
<pre><code>:::


## Calculate average methylation % per bin per region
The mean methylation of TSS regions was calculated across the entire 350 bp window. We used a sliding window approach to calculate mean methylation of gene bodies following (Laine et al., 2016). First, we subdivided the gene body into 40 bins (where the bin length therefore differs depending on the gene size), and then calculated the mean methylation for each bin, with an overlap between neighbouring bins of 250 bp. A similar sliding window approach was used to calculate mean methylation of up- and downstream regions. Here, the regions were divided into 40 bins of 250 bp each, and the mean methylation was calculated for each bin with an overlap between neighbouring bins of 125 bp. 


::: {.cell}

```{.r .cell-code  code-fold="true"}
#### Per gene, calculate meth% per bin
cpg_list &lt;- cpg_annotate %&gt;% group_split(gene_id, region)

save(cpg_list, file = "data/processed/annotated_cpg_list.RData")

# make function to calcalate average methylation per bin
meth_per_window &lt;- function(gene_region){tryCatch({
  #determine start and end of each region
 
    #determine window length for each region
  if (gene_region$region[1] == "TSS"){
    start_region &lt;- gene_region$start[1]
    end_region &lt;- gene_region$end[1]
    length = end_region - start_region
    overlap = 0
    } else if (gene_region$region[1] == "genes"){
    start_region &lt;- gene_region$start[1]
    end_region &lt;- gene_region$end[1]
    length = floor((end_region - start_region) / 40)
    overlap = 0.125*length 
    } else if (gene_region$region[1] == "downstream"){
    start_region &lt;- gene_region$start[1]
    end_region &lt;- gene_region$end[1]
    length = 250
    overlap = 125 
    } else if (gene_region$region[1] == "upstream"){
    start_region &lt;- gene_region$start[1]
    end_region &lt;- gene_region$end[1] 
    length = 250
    overlap = 125}
  
  ### function window slider adapted from R package windowscanr
  
  source("scripts/function_sliding_window_custom.R")
  window &lt;- win_scan(x = gene_region,
                     position = "pos",
                     values = "methperc",
                     win_size = length,
                     win_step = length,
                     overlap = overlap,
                     funs = c("mean", "sd"),
                     region = as.character(gene_region$region[1]))

  window[window == "NaN"] &lt;- NA
  sum_window &lt;- data.frame(window)
  sum_window$chr&lt;- gene_region$chr[1]
  sum_window$gene_id&lt;- gene_region$gene_id[1]
  sum_window$region&lt;- gene_region$region[1]
  sum_window$start_bp&lt;- start_region
  sum_window$end_bp&lt;- end_region
  sum_window$length_bin&lt;- length
  sum_window$win_nr &lt;- row.names(sum_window)
  
  return(sum_window)
  }, error = function(e){cat("ERROR :", conditionMessage(e), "\n");print(paste0(gene_region$chr[1], "_", gene_region$pos[1]))})}

# run in parallel
cpg_windows &lt;- parallel::mclapply(cpg_list, meth_per_window, mc.cores=12)
cpg_windows &lt;- do.call(rbind.data.frame, cpg_windows)

save(cpg_windows, file = "data/processed/cpg_meth_per_window_per_region.RData")</code></pre>
</section>
</section>
</div>
<section id="plot-the-result" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="plot-the-result"><span class="header-section-number">7.2</span> Plot the result</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### plotting</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>total_window <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">region =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"upstream"</span>, <span class="at">times =</span> <span class="dv">41</span>),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">rep</span>(<span class="st">"TSS"</span>, <span class="at">times =</span> <span class="dv">1</span>),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">rep</span>(<span class="st">"genes"</span>, <span class="at">times =</span> <span class="dv">41</span>),</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">rep</span>(<span class="st">"downstream"</span>, <span class="at">times =</span> <span class="dv">41</span>)),</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">win_nr =</span> <span class="fu">c</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">41</span>), <span class="dv">1</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">41</span>), <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">41</span>)),</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">window_total =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">124</span>))</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co">#extract for one gene</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="do">### first filter to have at least 2 cpg sites</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(cpg_windows, methperc_n <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">factor</span>(cpg_windows_min2<span class="sc">$</span>region, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"upstream"</span>, <span class="st">"TSS"</span>, <span class="st">"genes"</span>, <span class="st">"downstream"</span>))</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2<span class="sc">$</span>win_nr <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(cpg_windows_min2<span class="sc">$</span>win_nr)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2<span class="sc">$</span>methperc_mean <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(cpg_windows_min2<span class="sc">$</span>methperc_mean)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(cpg_windows_min2, total_window, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"win_nr"</span>, <span class="st">"region"</span>))</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(cpg_windows_min2, <span class="sc">!</span><span class="fu">is.na</span>(window_total))</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="co">#problem when gene is too short and don't have 41 windows</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_n <span class="ot">&lt;-</span> cpg_windows_min2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(window_total) <span class="sc">%&gt;%</span> </span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_meth =</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(methperc_mean)))</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_mean <span class="ot">&lt;-</span> cpg_windows_min2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(window_total) <span class="sc">%&gt;%</span> </span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_meth =</span> <span class="fu">mean</span>(methperc_mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a>            <span class="at">min_meth =</span> <span class="fu">min</span>(methperc_mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">max_meth =</span> <span class="fu">max</span>(methperc_mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_sd <span class="ot">&lt;-</span> cpg_windows_min2 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(window_total) <span class="sc">%&gt;%</span> </span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sd_meth =</span> <span class="fu">sd</span>(methperc_mean, <span class="at">na.rm=</span>T))</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_sum <span class="ot">&lt;-</span> <span class="fu">left_join</span>(cpg_windows_min2_n, cpg_windows_min2_mean, <span class="at">by =</span> <span class="st">"window_total"</span>) <span class="co">#leave out region for now, so wrong - talk to kees</span></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_sum <span class="ot">&lt;-</span> <span class="fu">left_join</span>(cpg_windows_min2_sum, cpg_windows_min2_sd, <span class="at">by =</span> <span class="st">"window_total"</span>)</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_sum<span class="sc">$</span>se_meth <span class="ot">&lt;-</span> cpg_windows_min2_sum<span class="sc">$</span>sd_meth<span class="sc">/</span><span class="fu">sqrt</span>(cpg_windows_min2_sum<span class="sc">$</span>n_meth)</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>cpg_windows_min2_sum[cpg_windows_min2_sum <span class="sc">==</span> <span class="st">"NaN"</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a><span class="co"># plot </span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a>clr_4 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#0F8D7BFF"</span>, <span class="st">"#928918FF"</span>, <span class="st">"#774CB3FF"</span>, <span class="st">"#436C97FF"</span>)</span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a>mean_tss_meth <span class="ot">&lt;-</span> <span class="fu">mean</span>(cpg_windows_min2<span class="sc">$</span>methperc_mean[<span class="fu">which</span>(cpg_windows_min2<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"TSS"</span>)], <span class="at">na.rm=</span>T)      </span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(cpg_windows_min2_sum, <span class="fu">aes</span>(<span class="at">x =</span> window_total, <span class="at">y =</span> mean_meth)) <span class="sc">+</span> </span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> mean_meth <span class="sc">-</span> se_meth, <span class="at">ymax =</span>mean_meth <span class="sc">+</span> se_meth), <span class="at">fill =</span> <span class="st">"lightblue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>)<span class="sc">+</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Mean CpG methylation %"</span>, <span class="at">title =</span> <span class="st">"CpG methylation across gene regions"</span>)<span class="sc">+</span></span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">label=</span><span class="st">"10 kb upstream"</span>, <span class="at">x =</span> <span class="dv">20</span>, <span class="at">y =</span> <span class="dv">66</span>, <span class="at">col =</span> clr_4[<span class="dv">1</span>], <span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">6</span>)<span class="sc">+</span></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">label=</span><span class="st">"Gene body"</span>, <span class="at">x =</span> <span class="dv">62</span>, <span class="at">y =</span> <span class="dv">66</span>, <span class="at">col =</span> clr_4[<span class="dv">3</span>], <span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">6</span>)<span class="sc">+</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">label=</span><span class="st">"10 kb downstream"</span>, <span class="at">x =</span> <span class="dv">104</span>, <span class="at">y =</span> <span class="dv">66</span>, <span class="at">col =</span> clr_4[<span class="dv">4</span>], <span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">6</span>)<span class="sc">+</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">label=</span><span class="st">"TSS"</span>, <span class="at">x =</span> <span class="dv">51</span>, <span class="at">y =</span> mean_tss_meth, <span class="at">col =</span> clr_4[<span class="dv">2</span>], <span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">6</span>)<span class="sc">+</span> </span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">xend =</span> <span class="dv">43</span>, <span class="at">y =</span> mean_tss_meth, <span class="at">x =</span> <span class="dv">47</span>, <span class="at">yend =</span> mean_tss_meth), <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.2</span>, <span class="st">"cm"</span>)))<span class="sc">+</span></span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">41.5</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> clr_4[<span class="dv">1</span>])<span class="sc">+</span> </span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">42.5</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> clr_4[<span class="dv">2</span>])<span class="sc">+</span> </span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">41.5</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> clr_4[<span class="dv">3</span>])<span class="sc">+</span> </span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">83.5</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> clr_4[<span class="dv">4</span>])<span class="sc">+</span> </span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>clr_4)<span class="sc">+</span></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">55</span>, <span class="dv">67</span>)<span class="sc">+</span></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="ot">-&gt;</span> across_genes_min2</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a>tss_min2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(cpg_windows_min2, region <span class="sc">==</span> <span class="st">"TSS"</span>)</span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a>tss_min2<span class="sc">$</span>methperc_n <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(tss_min2<span class="sc">$</span>methperc_n)</span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tss_min2, <span class="fu">aes</span>(<span class="at">x =</span> methperc_mean)) <span class="sc">+</span> <span class="fu">geom_histogram</span>() <span class="sc">+</span></span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Mean CpG methylation %"</span>, <span class="at">title =</span> <span class="st">"Histogram TSS CpG methylation % "</span>, <span class="at">subtitle =</span> <span class="st">"(min sites = 2)"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="ot">-&gt;</span> tss_hist_min2</span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tss_min2, <span class="fu">aes</span>(<span class="at">x =</span> methperc_n, <span class="at">y =</span> methperc_mean)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Number of CpGs"</span>, <span class="at">title =</span> <span class="st">"CpG sites vs mean methylation in TSS"</span>, <span class="at">subtitle =</span> <span class="st">"(min sites = 2)"</span>, <span class="at">y =</span> <span class="st">"Mean CpG methylation %"</span>) <span class="sc">+</span></span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>)<span class="ot">-&gt;</span> tss_n_vs_meth_min2</span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(tss_hist_min2, tss_n_vs_meth_min2, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">"hv"</span>, <span class="at">axis =</span> <span class="st">"bl"</span>) <span class="ot">-&gt;</span> tss_a_min2</span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(tss_a_min2, across_genes_min2, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">"h"</span>, <span class="at">axis =</span> <span class="st">"l"</span>) <span class="ot">-&gt;</span> tss_sum_min2</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(tss_sum_min2, <span class="at">file =</span> <span class="st">"plots/summary_wgbs_TSS_min2.png"</span>, <span class="at">width=</span><span class="dv">16</span>, <span class="at">height=</span><span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../plots/summary_wgbs_TSS_min2.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">CpG methylation across gene regions</figcaption>
</figure>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../qmd/1_intro_hypothesis.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction and hypotheses</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../qmd/3_explore.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Data exploration</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>