<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>The role of epigenetics in regulating life history trade-offs - 5&nbsp; Changing CpG sites</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../qmd/4_effort_meth.html" rel="next">
<link href="../qmd/2_filter.html" rel="prev">
<link href="../img/grouse_silhouette.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The role of epigenetics in regulating life history trade-offs</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../qmd/3_changing_meth.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Overview</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/0_intro_hypothesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction and hypotheses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/1_explore.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data exploration</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/2_filter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/3_changing_meth.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/4_effort_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/5_physio_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/6_fitness_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Fitness effects</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-change" id="toc-sec-change" class="nav-link active" data-scroll-target="#sec-change"><span class="header-section-number">5.1</span> Prepare the data</a></li>
  <li><a href="#function-to-run-the-model" id="toc-function-to-run-the-model" class="nav-link" data-scroll-target="#function-to-run-the-model"><span class="header-section-number">5.2</span> Function to run the model</a></li>
  <li><a href="#sec-changingcpg" id="toc-sec-changingcpg" class="nav-link" data-scroll-target="#sec-changingcpg"><span class="header-section-number">5.3</span> Run the model</a></li>
  <li><a href="#exclude-models-that-did-not-convergence" id="toc-exclude-models-that-did-not-convergence" class="nav-link" data-scroll-target="#exclude-models-that-did-not-convergence"><span class="header-section-number">5.4</span> Exclude models that did not convergence</a></li>
  <li><a href="#explore-and-filter-for-overdispersion" id="toc-explore-and-filter-for-overdispersion" class="nav-link" data-scroll-target="#explore-and-filter-for-overdispersion"><span class="header-section-number">5.5</span> Explore and filter for overdispersion</a></li>
  <li><a href="#fdr-correction" id="toc-fdr-correction" class="nav-link" data-scroll-target="#fdr-correction"><span class="header-section-number">5.6</span> FDR-correction</a></li>
  <li><a href="#filter-for-average-methylation-differences" id="toc-filter-for-average-methylation-differences" class="nav-link" data-scroll-target="#filter-for-average-methylation-differences"><span class="header-section-number">5.7</span> Filter for average methylation % differences</a></li>
  <li><a href="#plotting" id="toc-plotting" class="nav-link" data-scroll-target="#plotting"><span class="header-section-number">5.8</span> Plotting</a>
  <ul class="collapse">
  <li><a href="#volcano-and-manhattan-plots" id="toc-volcano-and-manhattan-plots" class="nav-link" data-scroll-target="#volcano-and-manhattan-plots"><span class="header-section-number">5.8.1</span> Volcano and Manhattan plots</a></li>
  <li><a href="#raw-data" id="toc-raw-data" class="nav-link" data-scroll-target="#raw-data"><span class="header-section-number">5.8.2</span> Raw data</a></li>
  </ul></li>
  <li><a href="#annotation" id="toc-annotation" class="nav-link" data-scroll-target="#annotation"><span class="header-section-number">5.9</span> Annotation</a>
  <ul class="collapse">
  <li><a href="#based-on-black-grouse-annotation" id="toc-based-on-black-grouse-annotation" class="nav-link" data-scroll-target="#based-on-black-grouse-annotation"><span class="header-section-number">5.9.1</span> Based on black grouse annotation</a></li>
  <li><a href="#gorilla" id="toc-gorilla" class="nav-link" data-scroll-target="#gorilla"><span class="header-section-number">5.9.2</span> GOrilla</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Identifying CpG sites that change over the lekking season</strong></p>
<p>Next, we want to identify CpG sites that significantly change in methylation level over the lekking season. To do so, we build a generalized linear mixed effect model per CpG site, using the time period (pre- or post-lekking) as a fixed effect.</p>
<p>Moreover, as there are (sometimes) multiple samples for each individual, we correct for the non-independence of individuals by including ID as a random effect, nested within lek. The response variables in these models are both the number of methylated and number of unmethylated C’s, using the <code>cbind</code> function.</p>
<section id="sec-change" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="sec-change"><span class="header-section-number">5.1</span> Prepare the data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">### load packages </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, data.table, tibble, performance, </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>               parallel, lmerTest, tidystats)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">### load epi data</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"data/processed/methylkit_prepost_long_onlyvar_thres0.3_min_0.5_group.RData"</span>) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="do">### load phenotype data </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/phenotypes/fulldata_complete_epi_withdates.RData"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>prepost <span class="ot">&lt;-</span> <span class="fu">subset</span>(all_pheno_epi, <span class="sc">!</span><span class="fu">is.na</span>(prepost)) <span class="co">#phenotype dataset also consists of other individuals not part of this study</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="do">### merge with metadata</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>prepost_long <span class="ot">&lt;-</span> <span class="fu">left_join</span>(prepost_long, prepost[,<span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"prepost"</span>, <span class="st">"Core"</span>, <span class="st">"year"</span>, <span class="st">"born"</span>, <span class="st">"site"</span>, <span class="st">"fulldate"</span>)], </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>                          <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"fulldate"</span>))</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="do">### add age data: core males have known full life histories, non-cores were not captured as yearlings</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>prepost_long <span class="ot">&lt;-</span> prepost_long <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">age_year =</span> <span class="fu">as.factor</span>(<span class="fu">case_when</span>(Core <span class="sc">==</span> <span class="st">"Core"</span> <span class="sc">~</span> year <span class="sc">-</span> born,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>                                                        Core <span class="sc">==</span> <span class="st">"No core"</span> <span class="sc">~</span> <span class="cn">NA</span>)),</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">age =</span> <span class="fu">as.factor</span>(<span class="fu">case_when</span>(Core <span class="sc">==</span> <span class="st">"Core"</span> <span class="sc">&amp;</span> (year <span class="sc">-</span> born <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">~</span> <span class="st">"Adult"</span>,</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>                                                        Core <span class="sc">==</span> <span class="st">"Core"</span> <span class="sc">&amp;</span> (year <span class="sc">-</span> born <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">~</span> <span class="st">"Yearling"</span>,</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>                                                        Core <span class="sc">==</span> <span class="st">"No core"</span> <span class="sc">~</span> <span class="st">"Adult"</span>)))</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="do">### intuitively, we would want to have "pre-lekking" as the reference level and "post-lekking" being the level compared to "pre", so let's swap these</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>prepost_long<span class="sc">$</span>prepost <span class="ot">&lt;-</span> <span class="fu">factor</span>(prepost_long<span class="sc">$</span>prepost, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"pre"</span>, <span class="st">"post"</span>))</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="do">### convert data to a list, one per CpG site</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> prepost_long <span class="sc">%&gt;%</span> <span class="fu">group_split</span>(chr_pos)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="function-to-run-the-model" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="function-to-run-the-model"><span class="header-section-number">5.2</span> Function to run the model</h2>
<p>Below you will find the full model used to 1) run the model in parallel per CpG site and 2) collect the output in a dataframe. Most importantly, this is the formula run for the model:</p>
<pre><code>lme4::glmer(cbind(numC, numT) ~ prepost + (1|id), family = "binomial")</code></pre>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">### define function to collect overdispersion statistics</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>overdisp.lmer_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  vpars <span class="ot">&lt;-</span> <span class="cf">function</span>(m) {</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">nrow</span>(m)<span class="sc">*</span>(<span class="fu">nrow</span>(m)<span class="sc">+</span><span class="dv">1</span>)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  model.df <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">VarCorr</span>(model),vpars))<span class="sc">+</span><span class="fu">length</span>(<span class="fu">fixef</span>(model))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  rdf <span class="ot">&lt;-</span> <span class="fu">nrow</span>(<span class="fu">model.frame</span>(model))<span class="sc">-</span>model.df</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  rp <span class="ot">&lt;-</span> <span class="fu">residuals</span>(model,<span class="at">type=</span><span class="st">"pearson"</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  Pearson.chisq <span class="ot">&lt;-</span> <span class="fu">sum</span>(rp<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  prat <span class="ot">&lt;-</span> Pearson.chisq<span class="sc">/</span>rdf</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  pval <span class="ot">&lt;-</span> <span class="fu">pchisq</span>(Pearson.chisq, <span class="at">df=</span>rdf, <span class="at">lower.tail=</span><span class="cn">FALSE</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">chisq=</span>Pearson.chisq,<span class="at">ratio=</span>prat,<span class="at">rdf=</span>rdf,<span class="at">p=</span>pval)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="do">### build function to run the model</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>function_model_glmer <span class="ot">&lt;-</span> <span class="cf">function</span>(df){<span class="fu">tryCatch</span>({</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  chr_pos <span class="ot">&lt;-</span> <span class="fu">as.character</span>(df[<span class="dv">1</span>,<span class="dv">1</span>])</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>prepost <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(df<span class="sc">$</span>prepost)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>id <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(df<span class="sc">$</span>id)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># model</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> lme4<span class="sc">::</span><span class="fu">glmer</span>(<span class="fu">cbind</span>(numC, numT) <span class="sc">~</span> prepost <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>id), <span class="at">family =</span> <span class="st">"binomial"</span>, df)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="co">#fixed effects</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  prepost_estimate <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>]</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  prepost_se <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>]</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  prepost_zval <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">3</span>]</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  prepost_pval <span class="ot">&lt;-</span>  <span class="fu">summary</span>(model)<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#random effects </span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  id_sd <span class="ot">&lt;-</span> <span class="fu">attributes</span>(<span class="fu">VarCorr</span>(model)<span class="sc">$</span>id)<span class="sc">$</span>stddev</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>  id_variance <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">VarCorr</span>(model), <span class="at">comp=</span><span class="st">"Variance"</span>)[<span class="dv">1</span>,<span class="dv">4</span>]</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  rsqc <span class="ot">&lt;-</span> performance<span class="sc">::</span><span class="fu">r2</span>(model)<span class="sc">$</span>R2_conditional <span class="co">#fixed plus random effects relative to overall variance</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  rsqm <span class="ot">&lt;-</span> performance<span class="sc">::</span><span class="fu">r2</span>(model)<span class="sc">$</span>R2_marginal <span class="co">#fixed effects relative to overall variance</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  dispersion.chisq <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">1</span>]</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>  dispersion.ratio <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">2</span>]</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  dispersion.rdf <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">3</span>]</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  dispersion.pval <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">4</span>]</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>  isSingular <span class="ot">&lt;-</span> <span class="fu">isSingular</span>(model)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">summary</span>(model)<span class="sc">$</span>optinfo<span class="sc">$</span>conv<span class="sc">$</span>lme4<span class="sc">$</span>messages )){</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>    convergence <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">summary</span>(model)<span class="sc">$</span>optinfo<span class="sc">$</span>conv<span class="sc">$</span>lme4<span class="sc">$</span>messages )){</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    convergence <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>optinfo<span class="sc">$</span>conv<span class="sc">$</span>lme4<span class="sc">$</span>messages</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  icc_id <span class="ot">&lt;-</span> <span class="fu">icc</span>(model, <span class="at">by_group =</span> <span class="cn">TRUE</span>, <span class="at">tolerance =</span> <span class="dv">0</span>)[<span class="dv">1</span>,<span class="dv">2</span>]</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">chr_pos=</span>chr_pos, </span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prepost_estimate =</span> prepost_estimate,</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prepost_se =</span> prepost_se,</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prepost_zval =</span> prepost_zval,</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prepost_pval =</span> prepost_pval,</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>                    <span class="at">id_sd =</span> id_sd,</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>                    <span class="at">id_variance =</span> id_variance,</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rsqc =</span> rsqc,</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rsqm =</span> rsqm,</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.chisq =</span> dispersion.chisq,</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.ratio =</span> dispersion.ratio,</span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.rdf =</span> dispersion.rdf,</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.pval =</span> dispersion.pval,</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>                    <span class="at">isSingular =</span> isSingular,</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>                    <span class="at">convergence =</span> convergence,</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>                    <span class="at">icc_id =</span> icc_id</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>                    ))</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>}, <span class="at">error =</span> <span class="cf">function</span>(e){<span class="fu">cat</span>(<span class="st">"ERROR :"</span>, <span class="fu">conditionMessage</span>(e), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>);<span class="fu">print</span>(chr_pos)})</span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="sec-changingcpg" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="sec-changingcpg"><span class="header-section-number">5.3</span> Run the model</h2>
<p>Here, we run the model in parallel, and next exclude CpG sites where there was overdispersion or convergence errors. After this filter step, we apply an FDR multiple-testing correction.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">### run the model in parallel per CpG site (list item)</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(data, function_model_glmer, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind.data.frame, out_glmer)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="exclude-models-that-did-not-convergence" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="exclude-models-that-did-not-convergence"><span class="header-section-number">5.4</span> Exclude models that did not convergence</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### Exclude models that did not converge ####</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>out_glmer_raw_conv <span class="ot">&lt;-</span> <span class="fu">subset</span>(out_glmer_raw, convergence <span class="sc">==</span> <span class="st">"boundary (singular) fit: see help('isSingular')"</span> <span class="sc">|</span> <span class="fu">is.na</span>(convergence))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(out_glmer_raw_conv) <span class="sc">/</span> <span class="fu">nrow</span>(out_glmer_raw) <span class="sc">*</span> <span class="dv">100</span> <span class="co"># retain 97.5%, 345937 out of 354765</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>When excluding models with convergence warnings/errors, we retain 97% of models, 345,983 out of 354,765 CpG site models.</p>
</section>
<section id="explore-and-filter-for-overdispersion" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="explore-and-filter-for-overdispersion"><span class="header-section-number">5.5</span> Explore and filter for overdispersion</h2>
<p>We can plot a histogram of the dispersion ratio and make a QQ plot to identify whether there is under/overdispersion and how to best filter for it.</p>
<p><img src="../plots/model_out/changing/hist_raw_glmer.png" class="img-fluid" alt="Histogram dispersion ratio and p-values raw models"> <img src="../plots/model_out/changing/qqplot_raw_glmer.png" class="img-fluid" alt="QQ plot model changing CpG sites raw models"></p>
<p>Next, we filter out the sites with an overdispersion ratio higher than the 95 percentile.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## filter for 95 percentile</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>out_glmer<span class="ot">&lt;-</span> <span class="fu">subset</span>(out_glmer_raw, dispersion.ratio <span class="sc">&lt;</span> <span class="fu">as.vector</span>(<span class="fu">quantile</span>(out_glmer_raw<span class="sc">$</span>dispersion.ratio, <span class="fl">0.95</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We are then left with a total of 328,640 CpG site models.</p>
<p><img src="../plots/model_out/changing/hist_filtered_glmer.png" class="img-fluid" alt="Histogram dispersion ratio and p-values filtered models"> <img src="../plots/model_out/changing/qqplot_raw_glmer.png" class="img-fluid" alt="QQ plot model changing CpG sites filtered models"></p>
</section>
<section id="fdr-correction" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="fdr-correction"><span class="header-section-number">5.6</span> FDR-correction</h2>
<p>After excluding CpG site models with overdispersion, we apply a multiple-testing correction using the false discovery rate (FDR).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>out_glmer<span class="sc">$</span>prepost_qval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(out_glmer<span class="sc">$</span>prepost_pval, <span class="at">method =</span> <span class="st">"fdr"</span>, <span class="at">n =</span> <span class="fu">nrow</span>(out_glmer))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filter-for-average-methylation-differences" class="level2" data-number="5.7">
<h2 data-number="5.7" class="anchored" data-anchor-id="filter-for-average-methylation-differences"><span class="header-section-number">5.7</span> Filter for average methylation % differences</h2>
<p>A “changing CpG site” is then defined as a CpG site with a significant effect of time period (FDR-corrected q-value &lt; 0.05) and at a mean difference in methylation % (across individuals) of at least -10% or 10%.</p>
<p>We first have to calculate delta methylation (post-lekking methylation % minus pre-lekking methylation %) in order to get this average.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Calculate delta methylation by matching up pre-post ####</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> <span class="fu">left_join</span>(<span class="fu">subset</span>(prepost_long, prepost <span class="sc">==</span> <span class="st">"pre"</span>),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">subset</span>(prepost_long, prepost <span class="sc">==</span> <span class="st">"post"</span>)[,<span class="fu">c</span>(<span class="st">"chr_pos"</span>, <span class="st">"lib_id"</span>, <span class="st">"epi_nr"</span>, <span class="st">"lib"</span>, <span class="st">"methperc"</span>, <span class="st">"cov"</span>, <span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"fulldate"</span>)],</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chr_pos"</span>, <span class="st">"id"</span>, <span class="st">"year"</span>), <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">"_pre"</span>, <span class="st">"_post"</span>))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(numC, numT, n_sample, prepost))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> <span class="fu">relocate</span>(<span class="fu">c</span>(id, year, born<span class="sc">:</span>age), <span class="at">.before=</span>lib_id_pre)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">delta_meth =</span> methperc_post <span class="sc">-</span> methperc_pre, <span class="at">.after =</span>born)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">diff_date =</span> fulldate_post <span class="sc">-</span> fulldate_pre)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>delta_meth<span class="sc">$</span>diff_date <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(delta_meth<span class="sc">$</span>diff_date)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co"># this file contains per CpG site the delta methylation levels per individual</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="do">### Calculate average delta_meth per CpG site across individuals</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>mean_delta_meth <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(chr_pos) <span class="sc">%&gt;%</span> <span class="fu">summarise_at</span>(<span class="fu">vars</span>(delta_meth), <span class="fu">funs</span>(<span class="fu">mean</span>(., <span class="at">na.rm=</span><span class="cn">TRUE</span>)))</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mean_delta_meth)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"mean_delta_meth"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Combine mean delta methylation with the model output</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> <span class="fu">left_join</span>(out_glmer, mean_delta_meth, <span class="at">by =</span> <span class="st">"chr_pos"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="do">### Filter min absolute mean methylation of 10%</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>sub_glmer_prepost <span class="ot">&lt;-</span> <span class="fu">subset</span>(out_glmer, prepost_qval <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(mean_delta_meth) <span class="sc">&gt;=</span> <span class="fl">0.1</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="do">### Save original data (per CpG site per individual) for models but only subset significant CpG sites</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>changing_cpg <span class="ot">&lt;-</span> <span class="fu">subset</span>(prepost_long, chr_pos <span class="sc">%in%</span> sub_glmer_prepost<span class="sc">$</span>chr_pos)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(changing_cpg, <span class="at">file=</span><span class="st">"results/modeloutput/changing/changing_sites_glmer.RData"</span>)</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="do">### Save the model output</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(out_glmer, <span class="at">file=</span><span class="st">"results/modeloutput/changing/modeloutput_glmer.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>A total of 1,026 CpG sites significantly change across the lekking period.</p>
</section>
<section id="plotting" class="level2" data-number="5.8">
<h2 data-number="5.8" class="anchored" data-anchor-id="plotting"><span class="header-section-number">5.8</span> Plotting</h2>
<section id="volcano-and-manhattan-plots" class="level3" data-number="5.8.1">
<h3 data-number="5.8.1" class="anchored" data-anchor-id="volcano-and-manhattan-plots"><span class="header-section-number">5.8.1</span> Volcano and Manhattan plots</h3>
<p>Next, we make a volcano plot to visualise the results.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> out_glmer <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sig =</span> <span class="fu">as.factor</span>(<span class="fu">case_when</span>(<span class="fu">abs</span>(mean_delta_meth) <span class="sc">&gt;=</span> <span class="fl">0.1</span> <span class="sc">&amp;</span> prepost_qval <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"sig"</span>, <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"nonsig"</span>)))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(out_glmer, <span class="fu">aes</span>(<span class="at">x =</span> mean_delta_meth, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">as.numeric</span>(prepost_qval)))) <span class="sc">+</span> </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">4</span>, <span class="at">alpha=</span><span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">col =</span> <span class="fu">as.factor</span>(sig))) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Mean "</span><span class="sc">*</span>Delta<span class="sc">*</span><span class="st">" methylation %"</span>), <span class="at">y =</span> <span class="st">"-log10(q-value)"</span>) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"grey60"</span>, clrs[<span class="dv">4</span>])) <span class="sc">+</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">col =</span> <span class="st">"darkred"</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="sc">-</span><span class="fl">0.1</span>, <span class="at">col =</span> <span class="st">"darkred"</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.1</span>, <span class="at">col =</span> <span class="st">"darkred"</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>) <span class="ot">-&gt;</span> volcano_change</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../plots/model_out/changing/volcano_change.png" class="img-fluid figure-img" width="450"></p>
<figcaption class="figure-caption">Volcano plot changing CpG sites</figcaption>
</figure>
</div>
<p>Interestingly, there seems to more CpG sites that increase in methylation over time compared to CpG sites that decrease in methylation over time (807 CpG sites increase, 219 decrease in methylation %).</p>
<p>We can also look at the distribution of CpG sites across the genome in a Manhattan plot.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/scaffold_names_dovetail.RData"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the chr_pos column into two columns based on the first "_"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>split_chr_pos <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(out_glmer<span class="sc">$</span>chr_pos, <span class="st">"_"</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the numbers following HRSCAF=XXX_number</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>out_glmer<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">1</span>), <span class="st">"_"</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">2</span>), <span class="st">";"</span>, </span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">4</span>), <span class="st">"="</span>,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">5</span>))</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>out_glmer<span class="sc">$</span>pos <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">6</span>))</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># join</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> <span class="fu">left_join</span>(out_glmer, genome[,<span class="fu">c</span>(<span class="st">"contig"</span>, <span class="st">"scaf_nr"</span>)], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"chr"</span> <span class="ot">=</span> <span class="st">"contig"</span>))</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a><span class="co"># plot </span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co"># lmer</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> <span class="fu">sample_n</span>(out_glmer, <span class="dv">100</span>)</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> out_glmer <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">col =</span> <span class="fu">case_when</span>(scaf_nr <span class="sc">%%</span> <span class="dv">2</span> <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"even"</span>,</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>                                        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"odd"</span>))</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>shade <span class="ot">&lt;-</span> out_glmer <span class="sc">%&gt;%</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset</span>(scaf_nr <span class="sc">&lt;=</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(scaf_nr) <span class="sc">%&gt;%</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">min =</span> <span class="fu">min</span>(pos), <span class="at">max =</span> <span class="fu">max</span>(pos)) <span class="sc">%&gt;%</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">min =</span> <span class="fu">case_when</span>(scaf_nr <span class="sc">==</span> <span class="dv">2</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">4</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">6</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">8</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">10</span> <span class="sc">|</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>                          scaf_nr <span class="sc">==</span> <span class="dv">12</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">14</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">16</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">18</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">20</span> <span class="sc">|</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>                          scaf_nr <span class="sc">==</span> <span class="dv">22</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">24</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">26</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">28</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">30</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> min)) <span class="sc">%&gt;%</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">max =</span> <span class="fu">case_when</span>(scaf_nr <span class="sc">==</span> <span class="dv">2</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">4</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">6</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">8</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">10</span> <span class="sc">|</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>                          scaf_nr <span class="sc">==</span> <span class="dv">12</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">14</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">16</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">18</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">20</span> <span class="sc">|</span></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>                          scaf_nr <span class="sc">==</span> <span class="dv">22</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">24</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">26</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">28</span> <span class="sc">|</span> scaf_nr <span class="sc">==</span> <span class="dv">30</span>  <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>                         <span class="cn">TRUE</span> <span class="sc">~</span> max))</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>                                        </span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>clrs <span class="ot">&lt;-</span> viridisLite<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">6</span>)</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="sc">%&gt;%</span> <span class="fu">subset</span>(scaf_nr <span class="sc">&lt;=</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pos, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">as.numeric</span>(prepost_pval)))) <span class="sc">+</span> </span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>, <span class="at">alpha=</span><span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">col =</span> <span class="fu">as.factor</span>(col), <span class="at">fill =</span> <span class="fu">as.factor</span>(col))) <span class="sc">+</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="sc">~</span>scaf_nr,<span class="at">scales =</span> <span class="st">'free_x'</span>, <span class="at">space =</span> <span class="st">'free_x'</span>, <span class="at">switch =</span> <span class="st">'x'</span>) <span class="sc">+</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Scaffold number"</span>, <span class="at">y =</span> <span class="fu">expression</span>(<span class="sc">-</span>log[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(p-value)"</span>)) <span class="sc">+</span></span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geom_rect(data=shade, aes(xmin=min, xmax=max, ymin=0, ymax=-log10(as.numeric(test$prepost_pval))), </span></span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>    <span class="co">#        alpha=0.5, fill = "#eceff4") + # "#f7f7f7" "#eceff4"</span></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>    <span class="co">#xlim(-1, 1)+</span></span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(clrs[<span class="dv">2</span>], clrs[<span class="dv">4</span>])) <span class="sc">+</span></span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">alpha</span>(<span class="fu">c</span>(clrs[<span class="dv">2</span>], clrs[<span class="dv">4</span>]), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span><span class="sc">/</span><span class="fu">nrow</span>(out_glmer)), <span class="at">col =</span> <span class="st">"darkred"</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">0</span>, <span class="st">"lines"</span>),</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="at">r =</span> <span class="fl">0.5</span>, <span class="at">l =</span> <span class="fl">0.1</span>, <span class="at">b =</span> <span class="fl">0.1</span>, <span class="at">t =</span> <span class="fl">0.1</span>, <span class="at">unit =</span> <span class="st">"cm"</span>),</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position=</span><span class="st">"none"</span>,</span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin=</span><span class="fu">margin</span>(<span class="at">t=</span><span class="dv">10</span>)),</span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin=</span><span class="fu">margin</span>(<span class="at">r=</span><span class="dv">5</span>)),</span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line.y =</span> <span class="fu">element_blank</span>()) <span class="ot">-&gt;</span> manhattan_change</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../plots/model_out/changing/manhattan_glmer.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Manhattan plot</figcaption>
</figure>
</div>
</section>
<section id="raw-data" class="level3" data-number="5.8.2">
<h3 data-number="5.8.2" class="anchored" data-anchor-id="raw-data"><span class="header-section-number">5.8.2</span> Raw data</h3>
<p>To assess whether the models performed well, I also plotted the raw data of the five most significantly changing CpG sites.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># raw plot for the 1st significant CpG site, repeat for the other 4 and combine with cowplot</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>out_glmer <span class="ot">&lt;-</span> out_glmer <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(prepost_qval)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">subset</span>(changing_cpg, chr_pos <span class="sc">==</span> out_glmer<span class="sc">$</span>chr_pos[<span class="dv">1</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(id, year) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> prepost, <span class="at">y =</span> methperc))<span class="sc">+</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">linewidth=</span><span class="dv">1</span>, <span class="at">outlier.shape=</span><span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="fu">aes</span>(<span class="at">group =</span> id_year), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"grey60"</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">seed =</span> <span class="dv">3922</span>)) <span class="sc">+</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size=</span>cov), <span class="at">col =</span> clrs[<span class="dv">4</span>], <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">seed =</span> <span class="dv">3922</span>)) <span class="sc">+</span> </span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Time period"</span>, <span class="at">y =</span> <span class="st">"Methylation percentage"</span>, <span class="at">subtitle =</span> out_glmer<span class="sc">$</span>chr_pos[<span class="dv">1</span>]) <span class="sc">+</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="quarto-figure quarto-figure-left">
<figure class="figure">
<p><img src="../plots/model_out/changing/rawdata_top_sig.png" class="img-fluid figure-img" width="350"></p>
<figcaption class="figure-caption">Raw data top 5 significantly changing CpG sites</figcaption>
</figure>
</div>
</section>
</section>
<section id="annotation" class="level2" data-number="5.9">
<h2 data-number="5.9" class="anchored" data-anchor-id="annotation"><span class="header-section-number">5.9</span> Annotation</h2>
<section id="based-on-black-grouse-annotation" class="level3" data-number="5.9.1">
<h3 data-number="5.9.1" class="anchored" data-anchor-id="based-on-black-grouse-annotation"><span class="header-section-number">5.9.1</span> Based on black grouse annotation</h3>
<p>Next, we annotate in which regions these changing CpG sites are located in based on the black grouse annotation.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">#### Annotation of changing CpG sites: gene regions #####</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="do">### Packages ####</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(genomation, GenomicFeatures, rtracklayer, </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>               GenomicRanges)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="do">### Combine all sites vs changing sites</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>cpg_all <span class="ot">&lt;-</span> out_glmer <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(chr_pos, prepost_qval))</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cpg_all)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"parameter_qval"</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>cpg_all<span class="sc">$</span>parameter <span class="ot">&lt;-</span> <span class="st">"all"</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>cpg_changing_select <span class="ot">&lt;-</span> changing_cpg <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(chr_pos, prepost_qval))</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cpg_changing_select)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"parameter_qval"</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>cpg_changing_select<span class="sc">$</span>parameter <span class="ot">&lt;-</span> <span class="st">"time_period"</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>all_models_sig <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cpg_all, cpg_changing_select)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="do">### Rename chr_pos and divide </span><span class="al">###</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"__"</span>, <span class="st">";"</span>, all_models_sig<span class="sc">$</span>chr_pos)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"HRSCAF_"</span>, <span class="st">"HRSCAF="</span>, all_models_sig<span class="sc">$</span>chr_pos, )</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the numbers following HRSCAF=XXX_number</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the chr_pos column into two columns based on the first "_"</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>split_chr_pos <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(all_models_sig<span class="sc">$</span>chr_pos, <span class="st">"_"</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">1</span>), <span class="st">"_"</span>,</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">2</span>))</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>pos <span class="ot">&lt;-</span> <span class="fu">sapply</span>(split_chr_pos, <span class="st">"["</span>, <span class="dv">3</span>)</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>all_models_sig <span class="ot">&lt;-</span> all_models_sig <span class="sc">%&gt;%</span> </span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(chr, <span class="at">.after =</span> chr_pos) <span class="sc">%&gt;%</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(pos, <span class="at">.after =</span> chr_pos)</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a><span class="co">#revert scafnames</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">";"</span>,<span class="st">"__"</span>, all_models_sig<span class="sc">$</span>chr_pos)</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr_pos <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"HRSCAF="</span>, <span class="st">"HRSCAF_"</span>, all_models_sig<span class="sc">$</span>chr_pos)</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">";"</span>,<span class="st">"__"</span>, all_models_sig<span class="sc">$</span>chr)</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>chr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"HRSCAF="</span>, <span class="st">"HRSCAF_"</span>, all_models_sig<span class="sc">$</span>chr)</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a><span class="do">### Load annotation data</span></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>annotation_dir <span class="ot">&lt;-</span> <span class="st">"~/PhD_grouse/grouse-annotation/output"</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>promoter<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/promoters.gff3"</span>)))</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>genes<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/genes.gff3"</span>)))</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>TSS<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/TSS.gff3"</span>)))</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>exons_gene<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/exons_gene.gff3"</span>)))</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>introns<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/introns_transcripts.gff3"</span>)))</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>downstream<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/downstream.gff3"</span>)))</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>upstream<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/upstream.gff3"</span>)))</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>threeUTR <span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/threeUTRs.gff3"</span>)))</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>fiveUTR<span class="ot">=</span><span class="fu">unique</span>(<span class="fu">gffToGRanges</span>(<span class="fu">paste0</span>(annotation_dir, <span class="st">"/fiveUTRs.gff3"</span>)))</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a><span class="do">#### Annotate ####</span></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>end <span class="ot">&lt;-</span> all_models_sig<span class="sc">$</span>pos</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>all_models_sig<span class="sc">$</span>start <span class="ot">&lt;-</span> all_models_sig<span class="sc">$</span>pos</span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a>sig_gr <span class="ot">&lt;-</span> <span class="fu">as</span>(all_models_sig, <span class="st">"GRanges"</span>)</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>sig_promoter <span class="ot">&lt;-</span> <span class="fu">subsetByOverlaps</span>(sig_gr, promoter) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"promoter"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>sig_gene <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, genes)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"gene"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a>sig_tss <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, TSS)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"TSS"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>sig_exon <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, exons_gene)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"exon"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a>sig_intron <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, introns))  <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"intron"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a>sig_down <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, downstream)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"downstream"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a>sig_up <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, upstream))  <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"upstream"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a>sig_threeUTR <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, threeUTR))  <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"threeUTR"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a>sig_fiveUTR <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">subsetByOverlaps</span>(sig_gr, fiveUTR))  <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="st">"region"</span> <span class="ot">=</span> <span class="st">"fiveUTR"</span>, <span class="at">.after=</span><span class="st">"parameter"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(seqnames<span class="sc">:</span>strand)) </span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a>all_models_sig_annotated <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sig_promoter, sig_gene,</span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a>                                  sig_tss, sig_exon, sig_intron, sig_down,</span>
<span id="cb13-99"><a href="#cb13-99" aria-hidden="true" tabindex="-1"></a>                                  sig_up, sig_threeUTR,  sig_fiveUTR)</span>
<span id="cb13-100"><a href="#cb13-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-102"><a href="#cb13-102" aria-hidden="true" tabindex="-1"></a><span class="do">#### Summarise number of sites per region ####</span></span>
<span id="cb13-103"><a href="#cb13-103" aria-hidden="true" tabindex="-1"></a>sum_annotated <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(<span class="fu">as.factor</span>(all_models_sig_annotated<span class="sc">$</span>region), all_models_sig_annotated<span class="sc">$</span>parameter))</span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sum_annotated) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"region"</span>, <span class="st">"model"</span>, <span class="st">"n"</span>)</span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>model <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"all"</span>, <span class="st">"All"</span>, sum_annotated<span class="sc">$</span>model)</span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>model <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"time_period"</span>, <span class="st">"Time period"</span>, sum_annotated<span class="sc">$</span>model)</span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-109"><a href="#cb13-109" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"downstream"</span>, <span class="st">"Downstream"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-110"><a href="#cb13-110" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"upstream"</span>, <span class="st">"Upstream"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-111"><a href="#cb13-111" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"exon"</span>, <span class="st">"Exon"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"fiveUTR"</span>, <span class="st">"5' UTR"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gene"</span>, <span class="st">"Gene body"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"intron"</span>, <span class="st">"Intron"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"promoter"</span>, <span class="st">"Promoter"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-116"><a href="#cb13-116" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"threeUTR"</span>, <span class="st">"3' UTR"</span>, sum_annotated<span class="sc">$</span>region)</span>
<span id="cb13-117"><a href="#cb13-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-118"><a href="#cb13-118" aria-hidden="true" tabindex="-1"></a>sum_annotated<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">factor</span>(sum_annotated<span class="sc">$</span>region, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"3' UTR"</span>, <span class="st">"5' UTR"</span>, <span class="st">"Downstream"</span>, <span class="st">"Upstream"</span>, <span class="st">"Gene body"</span>, <span class="st">"Exon"</span>, <span class="st">"Intron"</span>, <span class="st">"Promoter"</span>, <span class="st">"TSS"</span>))</span>
<span id="cb13-119"><a href="#cb13-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-120"><a href="#cb13-120" aria-hidden="true" tabindex="-1"></a><span class="co"># add total sig CpGs</span></span>
<span id="cb13-121"><a href="#cb13-121" aria-hidden="true" tabindex="-1"></a>sum_annotated <span class="ot">&lt;-</span> sum_annotated <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">n_total =</span> <span class="fu">case_when</span>(</span>
<span id="cb13-122"><a href="#cb13-122" aria-hidden="true" tabindex="-1"></a>  model <span class="sc">==</span> <span class="st">"All"</span> <span class="sc">~</span> <span class="dv">354649</span>,</span>
<span id="cb13-123"><a href="#cb13-123" aria-hidden="true" tabindex="-1"></a>  model <span class="sc">==</span> <span class="st">"Time period"</span> <span class="sc">~</span> <span class="fu">nrow</span>(cpg_change)))</span>
<span id="cb13-124"><a href="#cb13-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-125"><a href="#cb13-125" aria-hidden="true" tabindex="-1"></a>sum_annotated <span class="ot">&lt;-</span> sum_annotated <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">perc =</span> n <span class="sc">/</span> n_total <span class="sc">*</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<div class="cell-output-display">
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; overflow-x: scroll; width:99%; ">
<table class="lightable-classic-2 table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">region</th>
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">model</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">n</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">n_total</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">perc</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Downstream</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">134424</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">40.9031159</td>
</tr>
<tr class="even">
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">56072</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">17.0618306</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5' UTR</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">1531</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">0.4658593</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gene body</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">96490</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">29.3603944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">41272</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">12.5584226</td>
</tr>
<tr class="even">
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">83697</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">25.4676850</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3' UTR</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">279</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">0.0848953</td>
</tr>
<tr class="even">
<td style="text-align: left;">TSS</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">22620</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">6.8829114</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Upstream</td>
<td style="text-align: left;">All</td>
<td style="text-align: right;">151831</td>
<td style="text-align: right;">328640</td>
<td style="text-align: right;">46.1997931</td>
</tr>
<tr class="even">
<td style="text-align: left;">Downstream</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">402</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">39.1812865</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Exon</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">154</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">15.0097466</td>
</tr>
<tr class="even">
<td style="text-align: left;">5' UTR</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">0.8771930</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gene body</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">320</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">31.1890838</td>
</tr>
<tr class="even">
<td style="text-align: left;">Intron</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">169</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">16.4717349</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Promoter</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">171</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">16.6666667</td>
</tr>
<tr class="even">
<td style="text-align: left;">3' UTR</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">0.0974659</td>
</tr>
<tr class="odd">
<td style="text-align: left;">TSS</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">33</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">3.2163743</td>
</tr>
<tr class="even">
<td style="text-align: left;">Upstream</td>
<td style="text-align: left;">Time period</td>
<td style="text-align: right;">451</td>
<td style="text-align: right;">1026</td>
<td style="text-align: right;">43.9571150</td>
</tr>
</tbody>
</table>
</div>

</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../plots/model_out/changing/perc_sig_per_region.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Number of changing CpG sites per region</figcaption>
</figure>
</div>
</section>
<section id="gorilla" class="level3" data-number="5.9.2">
<h3 data-number="5.9.2" class="anchored" data-anchor-id="gorilla"><span class="header-section-number">5.9.2</span> GOrilla</h3>
<p>We next run the list of gene IDs of the significant sites as the target list as well as the list of gene IDs from all CpG sites that we ran the model on as a background list. I used default settings and the analysis was done based on a human model (<em>Homo sapiens</em>). The result can be seen below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>table_go_function_changing <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"../results/modeloutput/changing/GOFUNCTION_changing_similar.xlsx"</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>table_go_function_changing <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>Genes) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"^GO"</span>, table_go_function_changing<span class="sc">$</span><span class="st">`</span><span class="at">GO Term</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>() <span class="sc">%&gt;%</span>  <span class="fu">kable_classic_2</span>() <span class="sc">%&gt;%</span>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">"99%"</span>, <span class="at">height =</span> <span class="st">"200px"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; overflow-x: scroll; width:99%; ">
<table class="lightable-classic-2 table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">GO Term</th>
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">Description</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">P-value</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">FDR q-value</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">Enrichment</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">N</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">B</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">n</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">b</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0000987</td>
<td style="text-align: left;">proximal promoter sequence-specific DNA binding</td>
<td style="text-align: right;">5.63e-05</td>
<td style="text-align: right;">0.219</td>
<td style="text-align: right;">1.65</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">477</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">61</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0000978</td>
<td style="text-align: left;">RNA polymerase II proximal promoter sequence-specific DNA binding</td>
<td style="text-align: right;">1.26e-04</td>
<td style="text-align: right;">0.246</td>
<td style="text-align: right;">1.62</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">470</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">59</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:1990837</td>
<td style="text-align: left;">sequence-specific double-stranded DNA binding</td>
<td style="text-align: right;">1.42e-04</td>
<td style="text-align: right;">0.185</td>
<td style="text-align: right;">1.51</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">639</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">75</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0043565</td>
<td style="text-align: left;">sequence-specific DNA binding</td>
<td style="text-align: right;">1.54e-04</td>
<td style="text-align: right;">0.150</td>
<td style="text-align: right;">1.49</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">694</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">80</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0001012</td>
<td style="text-align: left;">RNA polymerase II regulatory region DNA binding</td>
<td style="text-align: right;">2.08e-04</td>
<td style="text-align: right;">0.162</td>
<td style="text-align: right;">1.55</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">541</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">65</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0000977</td>
<td style="text-align: left;">RNA polymerase II regulatory region sequence-specific DNA binding</td>
<td style="text-align: right;">2.08e-04</td>
<td style="text-align: right;">0.135</td>
<td style="text-align: right;">1.55</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">541</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">65</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0000976</td>
<td style="text-align: left;">transcription regulatory region sequence-specific DNA binding</td>
<td style="text-align: right;">2.13e-04</td>
<td style="text-align: right;">0.118</td>
<td style="text-align: right;">1.52</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">594</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0044212</td>
<td style="text-align: left;">transcription regulatory region DNA binding</td>
<td style="text-align: right;">2.35e-04</td>
<td style="text-align: right;">0.115</td>
<td style="text-align: right;">1.52</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">596</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0001067</td>
<td style="text-align: left;">regulatory region nucleic acid binding</td>
<td style="text-align: right;">2.47e-04</td>
<td style="text-align: right;">0.107</td>
<td style="text-align: right;">1.51</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">597</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">70</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0003700</td>
<td style="text-align: left;">DNA-binding transcription factor activity</td>
<td style="text-align: right;">3.51e-04</td>
<td style="text-align: right;">0.137</td>
<td style="text-align: right;">1.53</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">530</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">63</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0060228</td>
<td style="text-align: left;">phosphatidylcholine-sterol O-acyltransferase activator activity</td>
<td style="text-align: right;">4.64e-04</td>
<td style="text-align: right;">0.164</td>
<td style="text-align: right;">12.90</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0003690</td>
<td style="text-align: left;">double-stranded DNA binding</td>
<td style="text-align: right;">8.25e-04</td>
<td style="text-align: right;">0.268</td>
<td style="text-align: right;">1.43</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">688</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">76</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0005245</td>
<td style="text-align: left;">voltage-gated calcium channel activity</td>
<td style="text-align: right;">9.89e-04</td>
<td style="text-align: right;">0.297</td>
<td style="text-align: right;">4.11</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">7</td>
</tr>
</tbody>
</table>
</div>

</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>table_go_process_changing <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"../results/modeloutput/changing/GOPROCESS_changing_similar.xlsx"</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>table_go_process_changing <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>Genes) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"^GO"</span>, table_go_process_changing<span class="sc">$</span><span class="st">`</span><span class="at">GO Term</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>() <span class="sc">%&gt;%</span>  <span class="fu">kable_classic_2</span>() <span class="sc">%&gt;%</span>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">"99%"</span>, <span class="at">height =</span> <span class="st">"200px"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:200px; overflow-x: scroll; width:99%; ">
<table class="lightable-classic-2 table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">GO Term</th>
<th data-quarto-table-cell-role="th" style="text-align: left; position: sticky; top: 0; background-color: #FFFFFF;">Description</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">P-value</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">FDR q-value</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">Enrichment</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">N</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">B</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">n</th>
<th data-quarto-table-cell-role="th" style="text-align: right; position: sticky; top: 0; background-color: #FFFFFF;">b</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0045664</td>
<td style="text-align: left;">regulation of neuron differentiation</td>
<td style="text-align: right;">3.00e-07</td>
<td style="text-align: right;">0.00426</td>
<td style="text-align: right;">1.99</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">370</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">57</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:2000026</td>
<td style="text-align: left;">regulation of multicellular organismal development</td>
<td style="text-align: right;">1.10e-06</td>
<td style="text-align: right;">0.00772</td>
<td style="text-align: right;">1.50</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1067</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">124</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0050767</td>
<td style="text-align: left;">regulation of neurogenesis</td>
<td style="text-align: right;">2.30e-06</td>
<td style="text-align: right;">0.01030</td>
<td style="text-align: right;">1.81</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">450</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">63</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051094</td>
<td style="text-align: left;">positive regulation of developmental process</td>
<td style="text-align: right;">2.60e-06</td>
<td style="text-align: right;">0.00893</td>
<td style="text-align: right;">1.62</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">702</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">88</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0045597</td>
<td style="text-align: left;">positive regulation of cell differentiation</td>
<td style="text-align: right;">2.90e-06</td>
<td style="text-align: right;">0.00796</td>
<td style="text-align: right;">1.74</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">512</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">69</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051960</td>
<td style="text-align: left;">regulation of nervous system development</td>
<td style="text-align: right;">4.30e-06</td>
<td style="text-align: right;">0.00989</td>
<td style="text-align: right;">1.74</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">498</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">67</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0050793</td>
<td style="text-align: left;">regulation of developmental process</td>
<td style="text-align: right;">7.20e-06</td>
<td style="text-align: right;">0.01400</td>
<td style="text-align: right;">1.40</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1328</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">144</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0045666</td>
<td style="text-align: left;">positive regulation of neuron differentiation</td>
<td style="text-align: right;">8.20e-06</td>
<td style="text-align: right;">0.01390</td>
<td style="text-align: right;">2.18</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">207</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">35</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0022603</td>
<td style="text-align: left;">regulation of anatomical structure morphogenesis</td>
<td style="text-align: right;">1.45e-05</td>
<td style="text-align: right;">0.02200</td>
<td style="text-align: right;">1.63</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">587</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">74</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0060284</td>
<td style="text-align: left;">regulation of cell development</td>
<td style="text-align: right;">1.55e-05</td>
<td style="text-align: right;">0.02110</td>
<td style="text-align: right;">1.67</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">517</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">67</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0045595</td>
<td style="text-align: left;">regulation of cell differentiation</td>
<td style="text-align: right;">2.09e-05</td>
<td style="text-align: right;">0.02590</td>
<td style="text-align: right;">1.46</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">954</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">108</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0048522</td>
<td style="text-align: left;">positive regulation of cellular process</td>
<td style="text-align: right;">4.19e-05</td>
<td style="text-align: right;">0.04770</td>
<td style="text-align: right;">1.22</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">2823</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">266</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0050769</td>
<td style="text-align: left;">positive regulation of neurogenesis</td>
<td style="text-align: right;">5.89e-05</td>
<td style="text-align: right;">0.06180</td>
<td style="text-align: right;">1.93</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">254</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">38</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0010720</td>
<td style="text-align: left;">positive regulation of cell development</td>
<td style="text-align: right;">6.99e-05</td>
<td style="text-align: right;">0.06810</td>
<td style="text-align: right;">1.83</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">303</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">43</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:1903506</td>
<td style="text-align: left;">regulation of nucleic acid-templated transcription</td>
<td style="text-align: right;">7.98e-05</td>
<td style="text-align: right;">0.07260</td>
<td style="text-align: right;">1.30</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1645</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">166</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0006355</td>
<td style="text-align: left;">regulation of transcription, DNA-templated</td>
<td style="text-align: right;">7.98e-05</td>
<td style="text-align: right;">0.06810</td>
<td style="text-align: right;">1.30</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1645</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">166</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:2001141</td>
<td style="text-align: left;">regulation of RNA biosynthetic process</td>
<td style="text-align: right;">8.81e-05</td>
<td style="text-align: right;">0.07080</td>
<td style="text-align: right;">1.30</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1648</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">166</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0001525</td>
<td style="text-align: left;">angiogenesis</td>
<td style="text-align: right;">1.20e-04</td>
<td style="text-align: right;">0.09140</td>
<td style="text-align: right;">2.18</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">154</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">26</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0051004</td>
<td style="text-align: left;">regulation of lipoprotein lipase activity</td>
<td style="text-align: right;">1.31e-04</td>
<td style="text-align: right;">0.09380</td>
<td style="text-align: right;">6.45</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:1903508</td>
<td style="text-align: left;">positive regulation of nucleic acid-templated transcription</td>
<td style="text-align: right;">1.63e-04</td>
<td style="text-align: right;">0.11100</td>
<td style="text-align: right;">1.42</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">880</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">97</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0045893</td>
<td style="text-align: left;">positive regulation of transcription, DNA-templated</td>
<td style="text-align: right;">1.63e-04</td>
<td style="text-align: right;">0.10600</td>
<td style="text-align: right;">1.42</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">880</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">97</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0021781</td>
<td style="text-align: left;">glial cell fate commitment</td>
<td style="text-align: right;">1.68e-04</td>
<td style="text-align: right;">0.10400</td>
<td style="text-align: right;">10.32</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:1902680</td>
<td style="text-align: left;">positive regulation of RNA biosynthetic process</td>
<td style="text-align: right;">1.70e-04</td>
<td style="text-align: right;">0.10100</td>
<td style="text-align: right;">1.42</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">881</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">97</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051962</td>
<td style="text-align: left;">positive regulation of nervous system development</td>
<td style="text-align: right;">1.85e-04</td>
<td style="text-align: right;">0.10500</td>
<td style="text-align: right;">1.80</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">287</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">40</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0007399</td>
<td style="text-align: left;">nervous system development</td>
<td style="text-align: right;">1.92e-04</td>
<td style="text-align: right;">0.10500</td>
<td style="text-align: right;">2.05</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">28</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051239</td>
<td style="text-align: left;">regulation of multicellular organismal process</td>
<td style="text-align: right;">2.02e-04</td>
<td style="text-align: right;">0.10600</td>
<td style="text-align: right;">1.28</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1627</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">162</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048518</td>
<td style="text-align: left;">positive regulation of biological process</td>
<td style="text-align: right;">2.04e-04</td>
<td style="text-align: right;">0.10300</td>
<td style="text-align: right;">1.18</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">3168</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">289</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051252</td>
<td style="text-align: left;">regulation of RNA metabolic process</td>
<td style="text-align: right;">2.37e-04</td>
<td style="text-align: right;">0.11500</td>
<td style="text-align: right;">1.26</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1809</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">177</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0010628</td>
<td style="text-align: left;">positive regulation of gene expression</td>
<td style="text-align: right;">2.88e-04</td>
<td style="text-align: right;">0.13500</td>
<td style="text-align: right;">1.35</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1141</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">119</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0071526</td>
<td style="text-align: left;">semaphorin-plexin signaling pathway</td>
<td style="text-align: right;">3.00e-04</td>
<td style="text-align: right;">0.13700</td>
<td style="text-align: right;">4.30</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048523</td>
<td style="text-align: left;">negative regulation of cellular process</td>
<td style="text-align: right;">3.04e-04</td>
<td style="text-align: right;">0.13400</td>
<td style="text-align: right;">1.21</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">2480</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">232</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0031326</td>
<td style="text-align: left;">regulation of cellular biosynthetic process</td>
<td style="text-align: right;">3.11e-04</td>
<td style="text-align: right;">0.13200</td>
<td style="text-align: right;">1.24</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">2057</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">197</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">developmental process</td>
<td style="text-align: right;">3.33e-04</td>
<td style="text-align: right;">0.13800</td>
<td style="text-align: right;">1.20</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">2606</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">242</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0006357</td>
<td style="text-align: left;">regulation of transcription by RNA polymerase II</td>
<td style="text-align: right;">3.68e-04</td>
<td style="text-align: right;">0.14800</td>
<td style="text-align: right;">1.32</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1228</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">126</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0050770</td>
<td style="text-align: left;">regulation of axonogenesis</td>
<td style="text-align: right;">3.95e-04</td>
<td style="text-align: right;">0.15400</td>
<td style="text-align: right;">2.33</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">105</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">19</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0045944</td>
<td style="text-align: left;">positive regulation of transcription by RNA polymerase II</td>
<td style="text-align: right;">3.99e-04</td>
<td style="text-align: right;">0.15100</td>
<td style="text-align: right;">1.47</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">639</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">73</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0070830</td>
<td style="text-align: left;">bicellular tight junction assembly</td>
<td style="text-align: right;">4.07e-04</td>
<td style="text-align: right;">0.15000</td>
<td style="text-align: right;">3.75</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0120192</td>
<td style="text-align: left;">tight junction assembly</td>
<td style="text-align: right;">4.07e-04</td>
<td style="text-align: right;">0.14600</td>
<td style="text-align: right;">3.75</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0120193</td>
<td style="text-align: left;">tight junction organization</td>
<td style="text-align: right;">4.07e-04</td>
<td style="text-align: right;">0.14300</td>
<td style="text-align: right;">3.75</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">31</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0010975</td>
<td style="text-align: left;">regulation of neuron projection development</td>
<td style="text-align: right;">4.24e-04</td>
<td style="text-align: right;">0.14500</td>
<td style="text-align: right;">1.76</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">279</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">38</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0031328</td>
<td style="text-align: left;">positive regulation of cellular biosynthetic process</td>
<td style="text-align: right;">4.49e-04</td>
<td style="text-align: right;">0.15000</td>
<td style="text-align: right;">1.35</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1074</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">112</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0034371</td>
<td style="text-align: left;">chylomicron remodeling</td>
<td style="text-align: right;">4.64e-04</td>
<td style="text-align: right;">0.15100</td>
<td style="text-align: right;">12.90</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0010873</td>
<td style="text-align: left;">positive regulation of cholesterol esterification</td>
<td style="text-align: right;">4.64e-04</td>
<td style="text-align: right;">0.14700</td>
<td style="text-align: right;">12.90</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:1905048</td>
<td style="text-align: left;">regulation of metallopeptidase activity</td>
<td style="text-align: right;">4.73e-04</td>
<td style="text-align: right;">0.14700</td>
<td style="text-align: right;">8.60</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0033700</td>
<td style="text-align: left;">phospholipid efflux</td>
<td style="text-align: right;">4.73e-04</td>
<td style="text-align: right;">0.14300</td>
<td style="text-align: right;">8.60</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0034370</td>
<td style="text-align: left;">triglyceride-rich lipoprotein particle remodeling</td>
<td style="text-align: right;">4.73e-04</td>
<td style="text-align: right;">0.14000</td>
<td style="text-align: right;">8.60</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0009893</td>
<td style="text-align: left;">positive regulation of metabolic process</td>
<td style="text-align: right;">4.81e-04</td>
<td style="text-align: right;">0.14000</td>
<td style="text-align: right;">1.23</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1977</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">189</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0030154</td>
<td style="text-align: left;">cell differentiation</td>
<td style="text-align: right;">4.81e-04</td>
<td style="text-align: right;">0.13700</td>
<td style="text-align: right;">1.36</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1008</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">106</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0010896</td>
<td style="text-align: left;">regulation of triglyceride catabolic process</td>
<td style="text-align: right;">5.00e-04</td>
<td style="text-align: right;">0.13900</td>
<td style="text-align: right;">6.45</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0040008</td>
<td style="text-align: left;">regulation of growth</td>
<td style="text-align: right;">5.06e-04</td>
<td style="text-align: right;">0.13800</td>
<td style="text-align: right;">1.65</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">351</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">45</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0022604</td>
<td style="text-align: left;">regulation of cell morphogenesis</td>
<td style="text-align: right;">5.24e-04</td>
<td style="text-align: right;">0.14000</td>
<td style="text-align: right;">1.74</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">282</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">38</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0031325</td>
<td style="text-align: left;">positive regulation of cellular metabolic process</td>
<td style="text-align: right;">5.57e-04</td>
<td style="text-align: right;">0.14600</td>
<td style="text-align: right;">1.24</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1815</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">175</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0001558</td>
<td style="text-align: left;">regulation of cell growth</td>
<td style="text-align: right;">5.59e-04</td>
<td style="text-align: right;">0.14400</td>
<td style="text-align: right;">1.84</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">225</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">32</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0010557</td>
<td style="text-align: left;">positive regulation of macromolecule biosynthetic process</td>
<td style="text-align: right;">5.70e-04</td>
<td style="text-align: right;">0.14400</td>
<td style="text-align: right;">1.35</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1024</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">107</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0031323</td>
<td style="text-align: left;">regulation of cellular metabolic process</td>
<td style="text-align: right;">5.80e-04</td>
<td style="text-align: right;">0.14400</td>
<td style="text-align: right;">1.16</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">3137</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">283</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0051254</td>
<td style="text-align: left;">positive regulation of RNA metabolic process</td>
<td style="text-align: right;">6.42e-04</td>
<td style="text-align: right;">0.15700</td>
<td style="text-align: right;">1.36</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">937</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">99</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0045665</td>
<td style="text-align: left;">negative regulation of neuron differentiation</td>
<td style="text-align: right;">6.77e-04</td>
<td style="text-align: right;">0.16200</td>
<td style="text-align: right;">2.19</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">20</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0040011</td>
<td style="text-align: left;">locomotion</td>
<td style="text-align: right;">7.27e-04</td>
<td style="text-align: right;">0.17100</td>
<td style="text-align: right;">1.50</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">534</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">62</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048589</td>
<td style="text-align: left;">developmental growth</td>
<td style="text-align: right;">7.28e-04</td>
<td style="text-align: right;">0.16800</td>
<td style="text-align: right;">1.98</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">163</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">25</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0031113</td>
<td style="text-align: left;">regulation of microtubule polymerization</td>
<td style="text-align: right;">7.36e-04</td>
<td style="text-align: right;">0.16700</td>
<td style="text-align: right;">3.82</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048519</td>
<td style="text-align: left;">negative regulation of biological process</td>
<td style="text-align: right;">7.87e-04</td>
<td style="text-align: right;">0.17600</td>
<td style="text-align: right;">1.18</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">2777</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">253</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0048869</td>
<td style="text-align: left;">cellular developmental process</td>
<td style="text-align: right;">7.87e-04</td>
<td style="text-align: right;">0.17300</td>
<td style="text-align: right;">1.27</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1449</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">143</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0009891</td>
<td style="text-align: left;">positive regulation of biosynthetic process</td>
<td style="text-align: right;">8.10e-04</td>
<td style="text-align: right;">0.17600</td>
<td style="text-align: right;">1.32</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1091</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">112</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0019219</td>
<td style="text-align: left;">regulation of nucleobase-containing compound metabolic process</td>
<td style="text-align: right;">8.19e-04</td>
<td style="text-align: right;">0.17500</td>
<td style="text-align: right;">1.22</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">1997</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">189</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0009889</td>
<td style="text-align: left;">regulation of biosynthetic process</td>
<td style="text-align: right;">8.28e-04</td>
<td style="text-align: right;">0.17400</td>
<td style="text-align: right;">1.21</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">2106</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">198</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0015698</td>
<td style="text-align: left;">inorganic anion transport</td>
<td style="text-align: right;">9.42e-04</td>
<td style="text-align: right;">0.19500</td>
<td style="text-align: right;">2.45</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">79</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">15</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0040007</td>
<td style="text-align: left;">growth</td>
<td style="text-align: right;">9.55e-04</td>
<td style="text-align: right;">0.19500</td>
<td style="text-align: right;">1.94</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">166</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">25</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0030516</td>
<td style="text-align: left;">regulation of axon extension</td>
<td style="text-align: right;">9.95e-04</td>
<td style="text-align: right;">0.20000</td>
<td style="text-align: right;">2.77</td>
<td style="text-align: right;">8813</td>
<td style="text-align: right;">56</td>
<td style="text-align: right;">683</td>
<td style="text-align: right;">12</td>
</tr>
</tbody>
</table>
</div>

</div>
</div>
<p>Interestingly, there’s a lot of GO terms that are related to gene regulation - fitting the story perfectly! Moreover, the component that was highly significant is the “chylomicron”.</p>
<p>Wikipedia: “Chylomicrons transport lipids absorbed from the intestine to adipose, cardiac, and skeletal muscle tissue, where their triglyceride components are hydrolyzed by the activity of the lipoprotein lipase, allowing the released free fatty acids to be absorbed by the tissues. When a large portion of the triglyceride core has been hydrolyzed, chylomicron remnants are formed and are taken up by the liver, thereby also transferring dietary fat to the liver.”</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
      }
    }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../qmd/2_filter.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Filtering</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../qmd/4_effort_meth.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>