<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Epigenetic changes associated with reproductive investment in a lekking bird - 7&nbsp; Physio-associated CpG sites</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../qmd/6_fitness_meth.html" rel="next">
<link href="../qmd/4_effort_meth.html" rel="prev">
<link href="../img/grouse_silhouette.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Epigenetic changes associated with reproductive investment in a lekking bird</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../qmd/5_physio_meth.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/0_intro_hypothesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction and hypotheses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/1_explore.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data exploration</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/2_filter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/3_changing_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/4_effort_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/5_physio_meth.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/6_fitness_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Fitness effects</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-physio" id="toc-sec-physio" class="nav-link active" data-scroll-target="#sec-physio"><span class="header-section-number">7.1</span> Prepare the data</a></li>
  <li><a href="#function-to-run-the-model" id="toc-function-to-run-the-model" class="nav-link" data-scroll-target="#function-to-run-the-model"><span class="header-section-number">7.2</span> Function to run the model</a></li>
  <li><a href="#run-the-model-per-trait" id="toc-run-the-model-per-trait" class="nav-link" data-scroll-target="#run-the-model-per-trait"><span class="header-section-number">7.3</span> Run the model per trait</a>
  <ul class="collapse">
  <li><a href="#delta-mass" id="toc-delta-mass" class="nav-link" data-scroll-target="#delta-mass"><span class="header-section-number">7.3.1</span> Delta mass</a></li>
  <li><a href="#microfilaria-spp." id="toc-microfilaria-spp." class="nav-link" data-scroll-target="#microfilaria-spp."><span class="header-section-number">7.3.2</span> Microfilaria spp.</a></li>
  <li><a href="#trypanosoma-spp." id="toc-trypanosoma-spp." class="nav-link" data-scroll-target="#trypanosoma-spp."><span class="header-section-number">7.3.3</span> Trypanosoma spp.</a></li>
  <li><a href="#igg" id="toc-igg" class="nav-link" data-scroll-target="#igg"><span class="header-section-number">7.3.4</span> IgG</a></li>
  <li><a href="#hct" id="toc-hct" class="nav-link" data-scroll-target="#hct"><span class="header-section-number">7.3.5</span> HCT</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Which changes in CpG site methylation are associated with changes in physiology?</strong></p>
<p>Similar to the previous section, we next test which CpG site changes are significantly associated with changes in physiology. Physiological measures that we focus on include parasites (<em>Microfilaria spp.</em> and <em>Trypanosoma spp.</em>), body mass, IgG and HCT.</p>
<p>Again, we build a GLMM per CpG site to identify CpG sites whose DNA methylation change is associated with changes in physiology. For each CpG site for each trait, we build a model that predicts the changes DNA methylation, while fitting the change in physiology as a fixed effect. Here too, we run two models in parallel: one that does and one does that not include the fixed effect of pre-lekking methylation %.</p>
<section id="sec-physio" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="sec-physio"><span class="header-section-number">7.1</span> Prepare the data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">### load packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, data.table, tibble, </span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>               parallel, performance, lmerTest, tidystats)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="do">### load data</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file=</span><span class="st">"results/modeloutput/changing/changing_sites_glmer.RData"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="do">### load phenotypic data</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/phenotypes/fulldata_complete_epi_withdates.RData"</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/phenotypes/pheno_dif_prepost.RData"</span>)  <span class="do">## differences in physiology</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="do">### combine delta methylation data with site and behaviour info</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> <span class="fu">left_join</span>(delta_meth, <span class="fu">unique</span>(all_pheno_epi[,<span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"site"</span>, <span class="st">"Core"</span>)], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>)))</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="do">### z-transform the traits before the model</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>prepost_dif<span class="sc">$</span>mass_dif_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(prepost_dif<span class="sc">$</span>mass_dif)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>prepost_dif<span class="sc">$</span>microf_dif_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(prepost_dif<span class="sc">$</span>microf_dif)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>prepost_dif<span class="sc">$</span>trypa_dif_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(prepost_dif<span class="sc">$</span>trypa_dif)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>prepost_dif<span class="sc">$</span>ig_dif_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(prepost_dif<span class="sc">$</span>ig_dif)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>prepost_dif<span class="sc">$</span>hct_dif_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(prepost_dif<span class="sc">$</span>hct_dif)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="do">### combine data with pre-post delta physio numbers</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> <span class="fu">left_join</span>(delta_meth, <span class="fu">unique</span>(prepost_dif[,<span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"mass_dif"</span>, <span class="st">"microf_dif"</span>, <span class="st">"trypa_dif"</span>, <span class="st">"ig_dif"</span>, <span class="st">"hct_dif"</span>,</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"mass_dif_scl"</span>, <span class="st">"microf_dif_scl"</span>, <span class="st">"trypa_dif_scl"</span>, <span class="st">"ig_dif_scl"</span>, <span class="st">"hct_dif_scl"</span>)], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="function-to-run-the-model" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="function-to-run-the-model"><span class="header-section-number">7.2</span> Function to run the model</h2>
<p>Below you’ll find the function ran per trait per CpG site, where the LMM ran follows this structure:</p>
<p><code>lmerTest::lmer(delta_meth ~ scale(delta_trait) + methperc_pre + (1|site/id))</code></p>
<p>This model is identical to the one presented in <a href="4_effort_meth.html#sec-model-trait"><span>Section&nbsp;6.4</span></a>, except that the trait in question is also a calculation of the difference in the trait (delta; e.g.&nbsp;delta mass) rather than just one measure (e.g.&nbsp;attendance).</p>
</section>
<section id="run-the-model-per-trait" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="run-the-model-per-trait"><span class="header-section-number">7.3</span> Run the model per trait</h2>
<p>Here too, we only include CpG sites if they have at least 20 data points, which is therefore done for physiological trait separately. I’ll give the code for one trait only (delta mass) but the exact same is done for the rest.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">### run the model per trait</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/function_models.R"</span>) </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="do">### only select cpg sites with enough data</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>delta_meth_n_mass <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(chr_pos) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(delta_meth)<span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(mass_dif)) <span class="sc">%&gt;%</span> <span class="fu">tally</span>()</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>delta_meth_n_mass <span class="ot">&lt;-</span> <span class="fu">subset</span>(delta_meth_n_mass, n <span class="sc">&gt;</span> <span class="dv">20</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>delta_meth_sub_mass <span class="ot">&lt;-</span> <span class="fu">subset</span>(delta_meth, chr_pos <span class="sc">%in%</span> delta_meth_n_mass<span class="sc">$</span>chr_pos)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(delta_meth_sub_mass<span class="sc">$</span>chr_pos)) <span class="co">#808 sites</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>delta_meth_sub_mass_ls <span class="ot">&lt;-</span> delta_meth_sub_mass <span class="sc">%&gt;%</span> <span class="fu">group_split</span>(chr_pos)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="do">### mass</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="do">### no pre ####</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>m_mass_no_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_sub_mass_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"mass_dif"</span>, <span class="at">pre=</span><span class="st">"no_control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>m_mass_no_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_mass_no_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/physio/mass"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/physio"</span>,<span class="at">name_file =</span> <span class="st">"mass_dif_no_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Delta mass"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="do">### with pre ####</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>m_mass_with_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_sub_mass_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"mass_dif"</span>, <span class="at">pre=</span><span class="st">"control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>m_mass_with_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_mass_with_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/physio/mass"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/physio"</span>,<span class="at">name_file =</span> <span class="st">"mass_dif_with_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Delta mass"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="delta-mass" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="delta-mass"><span class="header-section-number">7.3.1</span> Delta mass</h3>
<p>808 CpG sites had enough phenotypic and epigenetic data to run the body mass model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking" class="level4" data-number="7.3.1.1">
<h4 data-number="7.3.1.1" class="anchored" data-anchor-id="with-pre-lekking"><span class="header-section-number">7.3.1.1</span> With pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta mass with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/mass/hist_raw_disp_pval_mass_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta mass with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/mass/volcawith_mass_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass with pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_with_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass with pre-methylation CpG 2</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_with_pre_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>CpG site “ScEsiA3_17616__HRSCAF_20466_3956821” is located in a downstream region (gene IDs ANN28634 similar to BPI and ANN28635 similar to ID1).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>BPI: BPI (Bactericidal Permeability Increasing Protein) is a Protein Coding gene. Diseases associated with BPI include Meningococcemia and Reactive Arthritis. Among its related pathways are Innate Immune System and Toll Like Receptor 3 (TLR3) Cascade. Gene Ontology (GO) annotations related to this gene include lipid binding and lipopolysaccharide binding. An important paralog of this gene is LBP.</p>
<p>ID1: The protein encoded by this gene is a helix-loop-helix (HLH) protein that can form heterodimers with members of the basic HLH family of transcription factors. The encoded protein has no DNA binding activity and therefore can inhibit the DNA binding and transcriptional activation ability of basic HLH proteins with which it interacts. This protein may play a role in cell growth, senescence, and differentiation. Two transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Jul 2008]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_18278__HRSCAF_21663_133779007” is located in a gene body, exon, upstream region and 5’UTR (gene IDs ANN06028 and ANN06030). This CpG site is the same that is significantly associated with attendance rate where we include pre-lekking methylation % in the model (!).</p>
</section>
<section id="without-pre-lekking" class="level4" data-number="7.3.1.2">
<h4 data-number="7.3.1.2" class="anchored" data-anchor-id="without-pre-lekking"><span class="header-section-number">7.3.1.2</span> Without pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta mass without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/mass/hist_raw_disp_pval_mass_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta mass without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/mass/volcano_mass_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass without pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_no_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass without pre-methylation CpG 2</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_no_pre_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass without pre-methylation CpG 3</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_no_pre_3.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass without pre-methylation CpG 4</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_no_pre_4.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta mass without pre-methylation CpG 5</figcaption>
<p><img src="../plots/model_out/physio/mass/mass_no_pre_5.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>One of the significant 5 CpG sites overlaps with the model that does include pre-methylation %.</p>
<p>CpG site “ScEsiA3_15486__HRSCAF_17393_53786697” is located in a gene body and intron (gene ID ANN20472 similar to Six4).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Six4: This gene encodes a member of the homeobox family, subfamily SIX. The drosophila homolog is a nuclear homeoprotein required for eye development. Studies in mouse show that this gene product functions as a transcription factor, and may have a role in the differentiation or maturation of neuronal cells. [provided by RefSeq, May 2010]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_17616__HRSCAF_20466_3956821” is located in a downstream region (gene ID ANN28634, similar to BPI, and ANN28635 similar to ID1). (see above)</p>
<p>CpG site “ScEsiA3_18752__HRSCAF_22883_4031831” is located in a gene body and intron (gene ID ANN31264 similar to TET3).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>TET3: Enables methyl-CpG binding activity and zinc ion binding activity. Involved in histone H3-K4 trimethylation; positive regulation of transcription by RNA polymerase II; and protein O-linked glycosylation. Predicted to be located in cytoplasm and male pronucleus. Predicted to be active in nucleus. Biomarker of esophagus squamous cell carcinoma. [provided by Alliance of Genome Resources, Apr 2022]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_21976__HRSCAF_26926_4403331” is located in a down- and upstream region (gene ID ANN30832, similar to Kiaa0319l).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Kiaa0319l: Predicted to act upstream of or within several processes, including flagellated sperm motility; proacrosomal vesicle fusion; and receptor-mediated endocytosis of virus by host cell. Located in Golgi apparatus; cytoplasmic vesicle; and nucleolus. [provided by Alliance of Genome Resources, Apr 2022]</p>
</div>
</div>
</div>
</section>
</section>
<section id="microfilaria-spp." class="level3" data-number="7.3.2">
<h3 data-number="7.3.2" class="anchored" data-anchor-id="microfilaria-spp."><span class="header-section-number">7.3.2</span> Microfilaria spp.</h3>
<p>149 CpG sites had enough phenotypic and epigenetic data to run the Microfilaria model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking-1" class="level4" data-number="7.3.2.1">
<h4 data-number="7.3.2.1" class="anchored" data-anchor-id="with-pre-lekking-1"><span class="header-section-number">7.3.2.1</span> With pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta Microfilaria spp. with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/microf/hist_raw_disp_pval_microf_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta Microfilaria spp. with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/microf/volcawith_microf_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta Microfilaria spp. with pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/microf/microf_with_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta Microfilaria spp. with pre-methylation CpG 2</figcaption>
<p><img src="../plots/model_out/physio/microf/microf_with_pre_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta Microfilaria spp. with pre-methylation CpG 3</figcaption>
<p><img src="../plots/model_out/physio/microf/microf_with_pre_3.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>CpG site “ScEsiA3_16771__HRSCAF_19097_2512626” is located in a gene body, exon and 5’UTR (gene ID ANN30965 similar to JAM3).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>JAM3: Tight junctions represent one mode of cell-to-cell adhesion in epithelial or endothelial cell sheets, forming continuous seals around cells and serving as a physical barrier to prevent solutes and water from passing freely through the paracellular space. The protein encoded by this immunoglobulin superfamily gene member is localized in the tight junctions between high endothelial cells. Unlike other proteins in this family, the this protein is unable to adhere to leukocyte cell lines and only forms weak homotypic interactions. The encoded protein is a member of the junctional adhesion molecule protein family and acts as a receptor for another member of this family. A mutation in an intron of this gene is associated with hemorrhagic destruction of the brain, subependymal calcification, and congenital cataracts. Alternative splicing results in multiple transcript variants.[provided by RefSeq, Apr 2011]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_21978__HRSCAF_26928_6268951” is located in a TSS, promoter (gene id ANN24632 similar to PPP1R1B) and upstream region (gene ID ANN24631 similar to STARD3).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>STARD3: This gene encodes a member of a subfamily of lipid trafficking proteins that are characterized by a C-terminal steroidogenic acute regulatory domain and an N-terminal metastatic lymph node 64 domain. The encoded protein localizes to the membranes of late endosomes and may be involved in exporting cholesterol. Alternative splicing results in multiple transcript variants.[provided by RefSeq, Oct 2009]</p>
</div>
</div>
</div>
<p>The third CpG site was unannotated.</p>
</section>
<section id="without-pre-lekking-1" class="level4" data-number="7.3.2.2">
<h4 data-number="7.3.2.2" class="anchored" data-anchor-id="without-pre-lekking-1"><span class="header-section-number">7.3.2.2</span> Without pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta Microfilaria spp. without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/microf/hist_raw_disp_pval_microf_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta Microfilaria spp. without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/microf/volcano_microf_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta Microfilaria spp. without pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/microf/microf_no_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>This is not the same site that was significant in the model with pre-lekking methylation.</p>
<p>This CpG site was unannotated.</p>
</section>
</section>
<section id="trypanosoma-spp." class="level3" data-number="7.3.3">
<h3 data-number="7.3.3" class="anchored" data-anchor-id="trypanosoma-spp."><span class="header-section-number">7.3.3</span> Trypanosoma spp.</h3>
<p>164 CpG sites had enough phenotypic and epigenetic data to run the Trypanosoma model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking-2" class="level4" data-number="7.3.3.1">
<h4 data-number="7.3.3.1" class="anchored" data-anchor-id="with-pre-lekking-2"><span class="header-section-number">7.3.3.1</span> With pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta Trypanosoma spp. with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/trypa/hist_raw_disp_pval_trypa_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta trypailaria spp. with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/trypa/volcawith_trypa_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="without-pre-lekking-2" class="level4" data-number="7.3.3.2">
<h4 data-number="7.3.3.2" class="anchored" data-anchor-id="without-pre-lekking-2"><span class="header-section-number">7.3.3.2</span> Without pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta trypailaria spp. without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/trypa/hist_raw_disp_pval_trypa_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta trypailaria spp. without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/trypa/volcano_trypa_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="igg" class="level3" data-number="7.3.4">
<h3 data-number="7.3.4" class="anchored" data-anchor-id="igg"><span class="header-section-number">7.3.4</span> IgG</h3>
<p>775 CpG sites had enough phenotypic and epigenetic data to run the IgG model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking-3" class="level4" data-number="7.3.4.1">
<h4 data-number="7.3.4.1" class="anchored" data-anchor-id="with-pre-lekking-3"><span class="header-section-number">7.3.4.1</span> With pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta IgG with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/ig/hist_raw_disp_pval_ig_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta IgG with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/ig/volcawith_ig_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta IgG with pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/ig/ig_with_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta IgG with pre-methylation CpG 2</figcaption>
<p><img src="../plots/model_out/physio/ig/ig_with_pre_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>Both CpG site “ScEsiA3_21979__HRSCAF_26929_1282292” (gene ID ANN27654 similar to NFIC) and “ScEsiA3_21979__HRSCAF_26929_1282301” (gene ID ANN27654 similar to NFIC) are located in upstream regions. The first one is the same site that was associated with centrality!</p>
</section>
<section id="without-pre-lekking-3" class="level4" data-number="7.3.4.2">
<h4 data-number="7.3.4.2" class="anchored" data-anchor-id="without-pre-lekking-3"><span class="header-section-number">7.3.4.2</span> Without pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta IgG without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/ig/hist_raw_disp_pval_ig_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta IgG without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/ig/volcano_ig_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta IgG without pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/ig/ig_no_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta IgG without pre-methylation CpG 2</figcaption>
<p><img src="../plots/model_out/physio/ig/ig_no_pre_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>These were the exact same CpG sites that are significant compared to the model that did include pre-methylation levels.</p>
</section>
</section>
<section id="hct" class="level3" data-number="7.3.5">
<h3 data-number="7.3.5" class="anchored" data-anchor-id="hct"><span class="header-section-number">7.3.5</span> HCT</h3>
<p>146 CpG sites had enough phenotypic and epigenetic data to run the HCT model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking-4" class="level4" data-number="7.3.5.1">
<h4 data-number="7.3.5.1" class="anchored" data-anchor-id="with-pre-lekking-4"><span class="header-section-number">7.3.5.1</span> With pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta HCT with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/hct/hist_raw_disp_pval_hct_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta HCT with pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/hct/volcawith_hct_dif_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: delta HCT with pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/physio/hct/hct_with_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>CpG site “ScEsiA3_15486__HRSCAF_17393_15878140” is located in a promoter and upstream region (gene ID ANN19643 similar to GAL: Galanin peptides (C. japonica)).</p>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>GAL: This gene encodes a neuroendocrine peptide that is widely expressed in the central and peripheral nervous systems and also the gastrointestinal tract, pancreas, adrenal gland and urogenital tract. The encoded protein is a precursor that is proteolytically processed to generate two mature peptides: galanin and galanin message-associated peptide (GMAP). Galanin has diverse physiological functions including nociception, feeding and energy homeostasis, osmotic regulation and water balance. GMAP has been demonstrated to possess antifungal activity and hypothesized to be part of the innate immune system. [provided by RefSeq, Jul 2015]</p>
</div>
</div>
</div>
</section>
<section id="without-pre-lekking-4" class="level4" data-number="7.3.5.2">
<h4 data-number="7.3.5.2" class="anchored" data-anchor-id="without-pre-lekking-4"><span class="header-section-number">7.3.5.2</span> Without pre-lekking</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: delta HCT without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/hct/hist_raw_disp_pval_hct_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: delta HCT without pre-methylation</figcaption>
<p><img src="../plots/model_out/physio/hct/volcano_hct_dif_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
      }
    }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../qmd/4_effort_meth.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../qmd/6_fitness_meth.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Fitness effects</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>