<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>The role of epigenetics in regulating life history trade-offs - 7&nbsp; Effort-associated CpG sites</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../qmd/7_physio_meth.html" rel="next">
<link href="../qmd/5_changing_meth.html" rel="prev">
<link href="../img/grouse_silhouette.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The role of epigenetics in regulating life history trade-offs</span>
    </a>
  </div>
        <div class="quarto-navbar-tools ms-auto">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../qmd/6_effort_meth.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Overview</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/1_intro_hypothesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Introduction and hypotheses</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/2_wgbs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Whole genome bisulphite sequencing</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/3_explore.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Data exploration</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/4_filter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Filtering</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/5_changing_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/6_effort_meth.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/7_physio_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/8_fitness_meth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Fitness effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../qmd/9_mutual_sites.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Mutual CpG sites</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#pre-methylation-vs-delta-methylation" id="toc-pre-methylation-vs-delta-methylation" class="nav-link active" data-scroll-target="#pre-methylation-vs-delta-methylation"><span class="header-section-number">7.1</span> Pre-methylation vs delta methylation</a></li>
  <li><a href="#sec-effort" id="toc-sec-effort" class="nav-link" data-scroll-target="#sec-effort"><span class="header-section-number">7.2</span> Prepare the data</a></li>
  <li><a href="#filter-for-n-20" id="toc-filter-for-n-20" class="nav-link" data-scroll-target="#filter-for-n-20"><span class="header-section-number">7.3</span> Filter for <em>n</em> &gt; 20</a></li>
  <li><a href="#sec-model-trait" id="toc-sec-model-trait" class="nav-link" data-scroll-target="#sec-model-trait"><span class="header-section-number">7.4</span> Function to run the model</a></li>
  <li><a href="#run-the-model-per-trait" id="toc-run-the-model-per-trait" class="nav-link" data-scroll-target="#run-the-model-per-trait"><span class="header-section-number">7.5</span> Run the model per trait</a>
  <ul class="collapse">
  <li><a href="#attendance" id="toc-attendance" class="nav-link" data-scroll-target="#attendance"><span class="header-section-number">7.5.1</span> Attendance</a></li>
  <li><a href="#fighting" id="toc-fighting" class="nav-link" data-scroll-target="#fighting"><span class="header-section-number">7.5.2</span> Fighting</a></li>
  <li><a href="#centrality" id="toc-centrality" class="nav-link" data-scroll-target="#centrality"><span class="header-section-number">7.5.3</span> Centrality</a></li>
  <li><a href="#without-pre-lekking-3" id="toc-without-pre-lekking-3" class="nav-link" data-scroll-target="#without-pre-lekking-3"><span class="header-section-number">7.5.4</span> Without pre-lekking</a></li>
  </ul></li>
  <li><a href="#concluding-remarks" id="toc-concluding-remarks" class="nav-link" data-scroll-target="#concluding-remarks"><span class="header-section-number">7.6</span> Concluding remarks</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Effort-associated CpG sites</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p><strong>Which changes in CpG site methylation are associated with reproductive effort?</strong></p>
<p>We have now identified a subset of CpG sites that significantly change over the lekking season, either by increasing or decreasing in methylation level. We next test which CpG site changes are significantly associated with different measures of reproductive effort.</p>
<p>Lekking is an energy-intensive, costly activity, as reflected by the losses in body mass in males that invest in reproduction (Lebigre et al., 2007). Therefore, we expect that the cost associated with high reproductive investment ledas to the up- or downregulation of associated genes, for example to invest more or less in certain other processess.</p>
<p>Similar to before, we build a GLMM per CpG site to identify CpG sites whose DNA methylation change is associated with reproductive effort. We quantify reproductive effort using three behavioural traits: lek attendance, fighting rate and lek centrality. For each CpG site for each trait, we build a model that predicts delta methylation, while fitting the behaviour as a fixed effect.</p>
<section id="pre-methylation-vs-delta-methylation" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="pre-methylation-vs-delta-methylation"><span class="header-section-number">7.1</span> Pre-methylation vs delta methylation</h2>
<p>Prior to the analysis, we investigated the relationship between delta methylation and pre-lekking methylation levels. We expect that pre-lekking methylation levels can affect delta methylation %. If pre-lekking methylation is 0%, there is potential for +100% change, but if pre-lekking methylation is 100%, there is potential for -100% change. On the other hand, if pre-lekking methylation is 50%, there is only potential for -50% and +50% change. Thus, we expect a U-shaped relationship between pre-lekking and absolute delta methylation. I plot the empirical data for this, based on only the CpG sites that significantly change across the lekking period. The red line in the left plot is the (linear) regression while correcting for id as a random effect (so a lmer output).</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Pre-lekking vs delta lekking methylation %</figcaption>
<p><img src="../plots/explore/pre_vs_delta.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>There does seem to be some association between the two, yet we might mask the effect of delta methylation on the phenotype if we include pre-lekking methylation in the model too. Therefore, we run two models in parallel: one that does include pre-lekking methylation % and one that does not, and then compare the model output / inference based on these two models. Note: for the manuscript we might want to choose for one in the main text and present the other in the supplements (although the exact CpG sites change, the conclusion that little is going on remains the same)</p>
</section>
<section id="sec-effort" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="sec-effort"><span class="header-section-number">7.2</span> Prepare the data</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">### load packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, data.table, tibble, gaston,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>               parallel, performance, lmerTest, tidystats, ggpointdensity)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>               </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/plotting_theme.R"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="do">### load data epi</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file=</span><span class="st">"results/modeloutput/changing/changing_sites_glmer.RData"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="do">### load phenotypic data</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"data/phenotypes/fulldata_complete_epi_withdates.RData"</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="do">### methylation difference</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"results/modeloutput/all_sites_deltameth.RData"</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> <span class="fu">subset</span>(delta_meth, chr_pos <span class="sc">%in%</span> changing_cpg<span class="sc">$</span>chr_pos)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="do">### combine delta methylation data with site and behaviour info</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> <span class="fu">left_join</span>(delta_meth, <span class="fu">unique</span>(all_pheno_epi[,<span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"site"</span>, <span class="st">"Core"</span>)], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>)))</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="do">### z-transform the traits before the model that subsets IDs and years where there is</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="do">### data for that CpG site</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>effort <span class="ot">&lt;-</span> all_pheno_epi <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"attend"</span>, <span class="st">"fight"</span>, <span class="st">"dist"</span>, <span class="st">"MS"</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(attend)) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>effort <span class="ot">&lt;-</span> <span class="fu">subset</span>(effort, id <span class="sc">%in%</span> delta_meth<span class="sc">$</span>id)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>effort<span class="sc">$</span>attend_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(effort<span class="sc">$</span>attend)</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>effort<span class="sc">$</span>fight_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(effort<span class="sc">$</span>fight)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>effort<span class="sc">$</span>dist_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(effort<span class="sc">$</span>dist)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="do">### combine reproductive effort data with methylation data</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>delta_meth <span class="ot">&lt;-</span> <span class="fu">left_join</span>(delta_meth, effort[,<span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>, <span class="st">"attend"</span>, <span class="st">"fight"</span>, <span class="st">"dist"</span>, <span class="st">"attend_scl"</span>, <span class="st">"fight_scl"</span>, <span class="st">"dist_scl"</span>)], <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"year"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Note that we z-transform the phenotypic data based on all samples, rather than on a model-by-model basis. This is because not all individuals will have epigenetic data for each CpG site, meaning that the z-transformed version of the phenotypic traits would change depending on the sample size, which we want to avoid.</p>
</section>
<section id="filter-for-n-20" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="filter-for-n-20"><span class="header-section-number">7.3</span> Filter for <em>n</em> &gt; 20</h2>
<p>Although we have filtered for CpG sites that have enough samples in both the pre- and post-lekking time periods, this does not mean that it was the same pair of samples that have data in both time periods. This means that we cannot outrule that we have a low sample size for delta methylation (where both pre- and post-lekking methylations for a given individual for a given CpG site had to be complete) differences that can be associated with phenotypic traits. Therefore, before we run the models, we filter for CpG sites that have at least 20 measures of delta methylation (and the phenotype). Since not all data is complete for all three behavioural traits, we do this on a trait-by-trait basis</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">### only select cpg sites with enough data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>delta_meth_n_dist <span class="ot">&lt;-</span> delta_meth <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(chr_pos) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(delta_meth)<span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(dist)) <span class="sc">%&gt;%</span> <span class="fu">tally</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>delta_meth_n_dist <span class="ot">&lt;-</span> <span class="fu">subset</span>(delta_meth_n_dist, n <span class="sc">&gt;</span> <span class="dv">20</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>delta_meth_sub_dist <span class="ot">&lt;-</span> <span class="fu">subset</span>(delta_meth, chr_pos <span class="sc">%in%</span> delta_meth_n_dist<span class="sc">$</span>chr_pos)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(delta_meth_sub_dist<span class="sc">$</span>chr_pos)) <span class="co"># 760 sites</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>delta_meth_dist_ls <span class="ot">&lt;-</span> delta_meth_sub_dist <span class="sc">%&gt;%</span> <span class="fu">group_split</span>(chr_pos)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># repeat for attendance and fighting rate</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>At this stage, we have the 1,026 CpG sites.</p>
</section>
<section id="sec-model-trait" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="sec-model-trait"><span class="header-section-number">7.4</span> Function to run the model</h2>
<p>You can find the full function below, but importantly this is the LMM formula I run for each CpG site for each trait:</p>
<p><code>lmgerTest::lmer(delta_meth ~ scale(trait) + methperc_pre + (1|site/id))</code></p>
<p>Within the function, we can define if we want to include pre-lekking methylation % or not by setting the ‘pre’ arument to “control” or “no_control” respectively.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># function to run the model</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>function_model_delta <span class="ot">&lt;-</span> <span class="cf">function</span>(df, parameter, pre){<span class="fu">tryCatch</span>({</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  chr_pos <span class="ot">&lt;-</span> <span class="fu">as.character</span>(df[<span class="dv">1</span>,<span class="dv">1</span>])</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>methperc_pre_scl <span class="ot">&lt;-</span> <span class="fu">scale</span>(df<span class="sc">$</span>methperc_pre)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (pre <span class="sc">==</span> <span class="st">"control"</span>){</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  formula <span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="fu">paste0</span>(<span class="st">"delta_meth ~ "</span>, parameter, <span class="st">"_scl + methperc_pre + (1|site/id) "</span>))}</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (pre <span class="sc">==</span> <span class="st">"no_control"</span>){</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  formula <span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="fu">paste0</span>(<span class="st">"delta_meth ~ "</span>, parameter, <span class="st">"_scl + (1|site/id) "</span>))}</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> lmerTest<span class="sc">::</span><span class="fu">lmer</span>(formula, <span class="at">data=</span>df)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  overdisp.lmer_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    vpars <span class="ot">&lt;-</span> <span class="cf">function</span>(m) {</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(m)<span class="sc">*</span>(<span class="fu">nrow</span>(m)<span class="sc">+</span><span class="dv">1</span>)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    model.df <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">VarCorr</span>(model),vpars))<span class="sc">+</span><span class="fu">length</span>(<span class="fu">fixef</span>(model))</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>    rdf <span class="ot">&lt;-</span> <span class="fu">nrow</span>(<span class="fu">model.frame</span>(model))<span class="sc">-</span>model.df</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>    rp <span class="ot">&lt;-</span> <span class="fu">residuals</span>(model,<span class="at">type=</span><span class="st">"pearson"</span>)</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>    Pearson.chisq <span class="ot">&lt;-</span> <span class="fu">sum</span>(rp<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>    prat <span class="ot">&lt;-</span> Pearson.chisq<span class="sc">/</span>rdf</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>    pval <span class="ot">&lt;-</span> <span class="fu">pchisq</span>(Pearson.chisq, <span class="at">df=</span>rdf, <span class="at">lower.tail=</span><span class="cn">FALSE</span>)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">chisq=</span>Pearson.chisq,<span class="at">ratio=</span>prat,<span class="at">rdf=</span>rdf,<span class="at">p=</span>pval)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  intercept <span class="ot">=</span> summary<span class="sc">$</span>coefficients[<span class="st">"(Intercept)"</span>, <span class="st">"Estimate"</span>]</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#fixed effect</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  parameter_estimate <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>]</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>  parameter_se <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>]</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  parameter_df <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">3</span>]</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  parameter_tval <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">4</span>]</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  parameter_pval <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">5</span>]</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (pre <span class="sc">==</span> <span class="st">"control"</span>){</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>  <span class="co">#premeth effect</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>    pre_estimate <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="st">"methperc_pre"</span>, <span class="st">"Estimate"</span>]</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>    pre_se <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="st">"methperc_pre"</span>, <span class="st">"Std. Error"</span>]</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    pre_df <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="st">"methperc_pre"</span>, <span class="st">"df"</span>]</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>    pre_tval <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="st">"methperc_pre"</span>, <span class="st">"t value"</span>]</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>    pre_pval <span class="ot">&lt;-</span> summary<span class="sc">$</span>coefficients[<span class="st">"methperc_pre"</span>, <span class="st">"Pr(&gt;|t|)"</span>]</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (pre <span class="sc">==</span> <span class="st">"no_control"</span>){</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>    pre_estimate <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>    pre_se <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    pre_df <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    pre_tval <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>    pre_pval <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  rsqc <span class="ot">&lt;-</span> performance<span class="sc">::</span><span class="fu">r2</span>(model)<span class="sc">$</span>R2_conditional <span class="co">#fixed plus random parameterects relative to overall variance</span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  rsqm <span class="ot">&lt;-</span> performance<span class="sc">::</span><span class="fu">r2</span>(model)<span class="sc">$</span>R2_marginal <span class="co">#fixed parameterects relative to overall variance</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>  dispersion.chisq <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">1</span>]</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>  dispersion.ratio <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">2</span>]</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>  dispersion.rdf <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">3</span>]</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>  dispersion.pval <span class="ot">&lt;-</span> <span class="fu">overdisp.lmer_fun</span>(model)[<span class="dv">1</span>,<span class="dv">4</span>]</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>  isSingular <span class="ot">&lt;-</span> <span class="fu">isSingular</span>(model)</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">summary</span>(model)<span class="sc">$</span>optinfo<span class="sc">$</span>conv<span class="sc">$</span>lme4<span class="sc">$</span>messages )){</span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>    convergence <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">summary</span>(model)<span class="sc">$</span>optinfo<span class="sc">$</span>conv<span class="sc">$</span>lme4<span class="sc">$</span>messages )){</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>    convergence <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>optinfo<span class="sc">$</span>conv<span class="sc">$</span>lme4<span class="sc">$</span>messages</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  icc_id_site <span class="ot">&lt;-</span><span class="fu">icc</span>(model, <span class="at">by_group =</span> <span class="cn">TRUE</span>, <span class="at">tolerance =</span> <span class="dv">0</span>)[<span class="dv">1</span>,<span class="dv">2</span>]</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>  icc_site <span class="ot">&lt;-</span><span class="fu">icc</span>(model, <span class="at">by_group =</span> <span class="cn">TRUE</span>, <span class="at">tolerance =</span> <span class="dv">0</span>)[<span class="dv">2</span>,<span class="dv">2</span>]</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">chr_pos=</span><span class="fu">as.factor</span>(chr_pos),</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parameter =</span> <span class="fu">as.factor</span>(parameter),</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>                    <span class="at">intercept =</span> <span class="fu">as.numeric</span>(intercept),</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>                    <span class="at">icc_id_site =</span> <span class="fu">as.numeric</span>(icc_id_site),</span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>                    <span class="at">icc_site =</span> <span class="fu">as.numeric</span>(icc_site),</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parameter_estimate =</span> <span class="fu">as.numeric</span>(parameter_estimate),</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parameter_se =</span> <span class="fu">as.numeric</span>(parameter_se),</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parameter_df =</span> <span class="fu">as.numeric</span>(parameter_df),</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parameter_tval =</span> <span class="fu">as.numeric</span>(parameter_tval),</span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>                    <span class="at">parameter_pval =</span> <span class="fu">as.numeric</span>(parameter_pval),</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pre_estimate =</span> <span class="fu">as.numeric</span>(pre_estimate),</span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pre_se =</span> <span class="fu">as.numeric</span>(pre_se),</span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pre_df =</span> <span class="fu">as.numeric</span>(pre_df),</span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pre_tval =</span> <span class="fu">as.numeric</span>(pre_tval),</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pre_pval =</span> <span class="fu">as.numeric</span>(pre_pval),</span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rsqc =</span> <span class="fu">as.numeric</span>(rsqc),</span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rsqm =</span> <span class="fu">as.numeric</span>(rsqm),</span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.chisq =</span> <span class="fu">as.numeric</span>(dispersion.chisq),</span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.ratio =</span> <span class="fu">as.numeric</span>(dispersion.ratio),</span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.rdf =</span> <span class="fu">as.numeric</span>(dispersion.rdf),</span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>                    <span class="at">dispersion.pval =</span> <span class="fu">as.numeric</span>(dispersion.pval),</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a>                    <span class="at">isSingular =</span> <span class="fu">as.logical</span>(isSingular),</span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>                    <span class="at">convergence =</span> convergence</span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>}, <span class="at">error =</span> <span class="cf">function</span>(e){<span class="fu">cat</span>(<span class="st">"ERROR :"</span>, <span class="fu">conditionMessage</span>(e), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>);<span class="fu">print</span>(<span class="fu">paste0</span>(chr_pos, <span class="st">" "</span>, <span class="fu">conditionMessage</span>(e)))})</span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-the-model-per-trait" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="run-the-model-per-trait"><span class="header-section-number">7.5</span> Run the model per trait</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"scripts/function_models.R"</span>) </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># the function function_model_delta_pheno runs the model</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># the function function_process_model contains a collector function that puts all data in a dataframe, excludes models with convergence errors,</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plots overdispersion, optionally filters for overdispersion (95 percentile), applies an FDR-correction,</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># makes a volcano plot, and saves all these data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In this function, we do the following (in this order):</p>
<p>*Convert the list into a dataframe <code>data &lt;- do.call(rbind.data.frame, list)</code></p>
<p>*Exclude models with convergence errors <code>data &lt;- subset(data, !grepl("converge", convergence))</code></p>
<p>*Create histograms of the overdispersion ratio and p-value <code>ggplot(data, aes(x = dispersion.ratio)) + geom_histogram()</code></p>
<p><code>ggplot(data, aes(x = parameter_pval))+ geom_histogram()</code></p>
<p>*Optionally exclude models with an overdispersion ratio higher than the 95 percentile (not done for any as there was no overdispersion)</p>
<p>*FDR correction <code>data$parameter_qval &lt;- p.adjust(data$parameter_pval, method = "fdr", n = nrow(data))</code></p>
<ul>
<li>Create a volcano plot where the model <em>beta</em> estimate is on the x-axis <code>ggplot(data, aes(x = parameter_estimate, y = -log10(parameter_qval))) + geom_point(aes(col = sig))</code></li>
</ul>
<section id="attendance" class="level3" data-number="7.5.1">
<h3 data-number="7.5.1" class="anchored" data-anchor-id="attendance"><span class="header-section-number">7.5.1</span> Attendance</h3>
<p>808 CpG sites had enough phenotypic and epigenetic data to run the attendance model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking" class="level4" data-number="7.5.1.1">
<h4 data-number="7.5.1.1" class="anchored" data-anchor-id="with-pre-lekking"><span class="header-section-number">7.5.1.1</span> With pre-lekking</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>m_attend_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_attend_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"attend"</span>, <span class="at">pre=</span><span class="st">"control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>m_attend_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_attend_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/effort"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/effort"</span>,</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">name_file =</span> <span class="st">"attend_with_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Attendance"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) <span class="co"># n sig = 3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="../plots/model_out/effort/hist_raw_disp_pval_attend_with_pre.png" class="img-fluid" alt="Histogram dispersion and p-value: attendance with pre-methylation"> <img src="../plots/model_out/effort/volcano_attend_with_pre.png" class="img-fluid" alt="Volcano plot: attendance with pre-methylation"> <img src="../plots/model_out/effort/raw/attend_with_pre_1.png" class="img-fluid" alt="Raw data: attendance with pre-methylation CpG 1"> <img src="../plots/model_out/effort/raw/attend_with_pre_2.png" class="img-fluid" alt="Raw data: attendance with pre-methylation CpG 2"> <img src="../plots/model_out/effort/raw/attend_with_pre_3.png" class="img-fluid" alt="Raw data: attendance with pre-methylation CpG 3"></p>
<p>CpG site “ScEsiA3_17655__HRSCAF_20552_371333” is located in an upstream region (gene ID ANN24887, similar to GPR87).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>GPR87: This gene encodes a G protein-coupled receptor and is located in a cluster of G protein-couple receptor genes on chromosome 3. The encoded protein has been shown to be overexpressed in lung squamous cell carcinoma (PMID:18057535) and regulated by p53 (PMID:19602589). [provided by RefSeq, Nov 2011]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_18278__HRSCAF_21663_133779007” is located in the exon of ANN06029 (similar to Adck2) and upstream from ANN06028 (closer) and ANN06030 (both unknown functions). This is the same CpG site that is associated with changes in body mass (when including pre-lekking meth %)!</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Adck2: This gene AarF Domain Containing Kinase 2 is predicted to enable ATP binding activity and protein serine/threonine kinase activity. Predicted to be involved in protein phosphorylation. Predicted to be integral component of membrane. [provided by Alliance of Genome Resources, Apr 2022]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_18752__HRSCAF_22883_2574288” is located in upstream from gene ID ANN31167 (similar to PPP2R2A) and downstream from gene ID ANN31166 (unknown function, closer).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>PPP2R2A: The product of this gene belongs to the phosphatase 2 regulatory subunit B family. Protein phosphatase 2 is one of the four major Ser/Thr phosphatases, and it is implicated in the negative control of cell growth and division. It consists of a common heteromeric core enzyme, which is composed of a catalytic subunit and a constant regulatory subunit, that associates with a variety of regulatory subunits. The B regulatory subunit might modulate substrate selectivity and catalytic activity. This gene encodes an alpha isoform of the regulatory subunit B55 subfamily. Alternatively spliced transcript variants have been described. [provided by RefSeq, Apr 2010]</p>
</div>
</div>
</div>
</section>
<section id="without-pre-lekking" class="level4" data-number="7.5.1.2">
<h4 data-number="7.5.1.2" class="anchored" data-anchor-id="without-pre-lekking"><span class="header-section-number">7.5.1.2</span> Without pre-lekking</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>m_attend_no_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"attend"</span>, <span class="at">pre=</span><span class="st">"no_control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>m_attend_no_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_attend_no_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/effort"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/effort"</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">name_file =</span> <span class="st">"attend_no_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Attendance"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) <span class="co"># n sig = 0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: attendance no pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/hist_raw_disp_pval_attend_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: attendance no pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/volcano_attend_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="fighting" class="level3" data-number="7.5.2">
<h3 data-number="7.5.2" class="anchored" data-anchor-id="fighting"><span class="header-section-number">7.5.2</span> Fighting</h3>
<p>466 CpG sites had enough phenotypic and epigenetic data to run the fighting model (after filtering for n &gt; 20).</p>
<section id="without-pre-lekking-1" class="level4" data-number="7.5.2.1">
<h4 data-number="7.5.2.1" class="anchored" data-anchor-id="without-pre-lekking-1"><span class="header-section-number">7.5.2.1</span> Without pre-lekking</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>m_fight_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_fight_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"fight"</span>, <span class="at">pre=</span><span class="st">"control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>m_fight_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_fight_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/effort"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/effort"</span>,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">name_file =</span> <span class="st">"fight_with_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Centrality"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) <span class="co"># n sig = 3</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: fighting rate with pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/hist_raw_disp_pval_fight_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: fighting rate with pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/volcano_fight_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: fighting rate with pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/effort/raw/fight_with_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: fighting rate with pre-methylation CpG 2</figcaption>
<p><img src="../plots/model_out/effort/raw/fight_with_pre_2.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: fighting rate with pre-methylation CpG 3</figcaption>
<p><img src="../plots/model_out/effort/raw/fight_with_pre_3.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>CpG site “ScEsiA3_15486__HRSCAF_17393_6262304” is located in the exon of gene ID ANN19413 (similar to RHOF) and upstream from gene ID ANN19414 (similar to Rad9a).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>RHOF: Ras Homolog Family Member F, Filopodia Associated. Predicted to enable GTP binding activity; GTPase activity; and protein kinase binding activity. Involved in actin filament organization. Located in extracellular exosome. [provided by Alliance of Genome Resources, Apr 2022]</p>
<p>Rad9a: This gene product is highly similar to Schizosaccharomyces pombe rad9, a cell cycle checkpoint protein required for cell cycle arrest and DNA damage repair. This protein possesses 3’ to 5’ exonuclease activity, which may contribute to its role in sensing and repairing DNA damage. It forms a checkpoint protein complex with RAD1 and HUS1. This complex is recruited by checkpoint protein RAD17 to the sites of DNA damage, which is thought to be important for triggering the checkpoint-signaling cascade. Alternatively spliced transcript variants encoding different isoforms have been found for this gene. [provided by RefSeq, Aug 2011]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_18290__HRSCAF_21698_5473558” is located in an upstream region (gene ID ANN29135, similar to SLC38A10).</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>SLC38A10: Predicted to enable amino acid transmembrane transporter activity. Predicted to be involved in amino acid transmembrane transport. Predicted to act upstream of or within bone development. Located in Golgi apparatus. [provided by Alliance of Genome Resources, Apr 2022]</p>
</div>
</div>
</div>
<p>CpG site “ScEsiA3_21978__HRSCAF_26928_5532732” is located in an intronic region of gene ID ANN24560 (similar to Jup) and upstream region of gene IDs ANN24559 (similar to TRAK1) and ANN24561 similar to Fkbp10.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>JUP: This gene encodes a major cytoplasmic protein which is the only known constituent common to submembranous plaques of both desmosomes and intermediate junctions. This protein forms distinct complexes with cadherins and desmosomal cadherins and is a member of the catenin family since it contains a distinct repeating amino acid motif called the armadillo repeat. Mutation in this gene has been associated with Naxos disease. Alternative splicing occurs in this gene; however, not all transcripts have been fully described. [provided by RefSeq, Jul 2008]</p>
<p>TRAK1: Predicted to enable GABA receptor binding activity and myosin binding activity. Involved in endosome to lysosome transport. Located in early endosome and mitochondrion. Implicated in developmental and epileptic encephalopathy 68. [provided by Alliance of Genome Resources, Apr 2022]</p>
<p>Fkbp10: The protein encoded by this gene belongs to the FKBP-type peptidyl-prolyl cis/trans isomerase (PPIase) family. This protein localizes to the endoplasmic reticulum and acts as a molecular chaperone. Alternatively spliced variants encoding different isoforms have been reported, but their biological validity has not been determined.[provided by RefSeq, Nov 2009]</p>
</div>
</div>
</div>
</section>
<section id="without-pre-lekking-2" class="level4" data-number="7.5.2.2">
<h4 data-number="7.5.2.2" class="anchored" data-anchor-id="without-pre-lekking-2"><span class="header-section-number">7.5.2.2</span> Without pre-lekking</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>m_fight_no_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"fight"</span>, <span class="at">pre=</span><span class="st">"no_control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>m_fight_no_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_fight_no_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/effort"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/effort"</span>,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">name_file =</span> <span class="st">"fight_no_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Fighting"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) <span class="co"># 0 sig</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: fighting rate no pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/hist_raw_disp_pval_fight_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: fighting rate no pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/volcano_fight_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="centrality" class="level3" data-number="7.5.3">
<h3 data-number="7.5.3" class="anchored" data-anchor-id="centrality"><span class="header-section-number">7.5.3</span> Centrality</h3>
<p>760 CpG sites had enough phenotypic and epigenetic data to run the centrality model (after filtering for n &gt; 20).</p>
<section id="with-pre-lekking-1" class="level4" data-number="7.5.3.1">
<h4 data-number="7.5.3.1" class="anchored" data-anchor-id="with-pre-lekking-1"><span class="header-section-number">7.5.3.1</span> With pre-lekking</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>m_dist_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"dist"</span>, <span class="at">pre=</span><span class="st">"control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>m_dist_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_dist_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/effort"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/effort"</span>,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">name_file =</span> <span class="st">"dist_with_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Centrality"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) <span class="co"># n sig = 1 </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: centrality with pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/hist_raw_disp_pval_dist_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: centrality with pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/volcano_dist_with_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: centrality with pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/effort/raw/dist_with_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>CpG site “ScEsiA3_21979__HRSCAF_26929_1282292” is located in an upstream region (gene ID ANN27654 similar to NFIC). Note that this is the same site that is significantly associated with delta IgG!!</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>NFIC: The protein encoded by this gene belongs to the CTF/NF-I family. These are dimeric DNA-binding proteins, and function as cellular transcription factors and as replication factors for adenovirus DNA replication. Alternatively spliced transcript variants encoding different isoforms have been described for this gene. [provided by RefSeq, Oct 2011]</p>
</div>
</div>
</div>
</section>
</section>
<section id="without-pre-lekking-3" class="level3" data-number="7.5.4">
<h3 data-number="7.5.4" class="anchored" data-anchor-id="without-pre-lekking-3"><span class="header-section-number">7.5.4</span> Without pre-lekking</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>m_dist_no_pre <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(delta_meth_ls, function_model_delta_pheno, <span class="at">parameter=</span><span class="st">"dist"</span>, <span class="at">pre=</span><span class="st">"no_control"</span>, <span class="at">mc.cores=</span><span class="dv">4</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>m_dist_no_pre_out <span class="ot">&lt;-</span> <span class="fu">function_process_model</span>(m_dist_no_pre, <span class="at">dir_plots =</span> <span class="st">"plots/model_out/effort"</span>, <span class="at">dir_data =</span> <span class="st">"results/modeloutput/effort"</span>,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">name_file =</span> <span class="st">"dist_no_pre"</span>, <span class="at">pretty_name =</span> <span class="st">"Centrality"</span>, <span class="at">filter_disp=</span><span class="cn">FALSE</span>) <span class="co"># n sig = 1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This CpG site is the same one that was significant as the model run with pre-methylation.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Histogram dispersion and p-value: centrality no pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/hist_raw_disp_pval_dist_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Volcano plot: centrality no pre-methylation</figcaption>
<p><img src="../plots/model_out/effort/volcano_dist_no_pre.png" class="img-fluid figure-img"></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<figcaption class="figure-caption">Raw data: centrality no pre-methylation CpG 1</figcaption>
<p><img src="../plots/model_out/effort/raw/dist_no_pre_1.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>This is the same CpG site that was significant in the model with pre-lekking methylation %</p>
</section>
</section>
<section id="concluding-remarks" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="concluding-remarks"><span class="header-section-number">7.6</span> Concluding remarks</h2>
<p>So it seems like very litte CpG methylation changes are associated with reproductive effort. When we do not control for pre-lekking methylation %, we don’t find anything significant for attendance and fighting rate, and we find the same single CpG significant for centrality that is significant in the model where we do control for pre-lekking methylation. We find 3, 3, and 1 significant CpG site for lek attendance, fighting rate and centrality respectively in the models where we do control for pre-lekking methylation.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
      }
    }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../qmd/5_changing_meth.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Changing CpG sites</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../qmd/7_physio_meth.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Physio-associated CpG sites</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>